{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "11ad24f9-e48e-4f22-bbe3-6360e55f70a4",
   "metadata": {},
   "source": [
    "# **ClinVar VCF Parser**\n",
    "### Author: Daniel Brock\n",
    "### Date: 7/1/2023\n",
    "### Purpose: to select pathogenic and likely pathogenic (PLP) and benign and likely benign (BLB) SNVs from ClinVar after 2020-11-21 (date of MutScore paper cutoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "51b9f321-b0d0-47e3-9525-330c3df8c114",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\TooFastDan\\OneDrive - Baylor College of Medicine\\BCM\\Projects\\Autosomal Dominant Predictor of IRDs\\manuscript\\GitHub\n"
     ]
    }
   ],
   "source": [
    "# Importing required packages\n",
    "import io\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "# Working directory\n",
    "cwd = os.getcwd()\n",
    "print(cwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "8fdd7e53-918f-461b-af6e-d37ecf6dd17c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def read_vcf(path):\n",
    "    \"\"\"\n",
    "    Function to parse a vcf file as a dataframe.\n",
    "    Source: https://gist.github.com/dceoy/99d976a2c01e7f0ba1c813778f9db744\n",
    "    Input: a file path to a VCF file from ClinVar\n",
    "    Output: a pandas dataframe\n",
    "    \"\"\"\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(\n",
    "        io.StringIO(''.join(lines)),\n",
    "        dtype={'#CHROM': str, 'POS': int, 'ID': str, 'REF': str, 'ALT': str, 'QUAL': str, 'FILTER': str, 'INFO': str},\n",
    "        sep='\\t'\n",
    "    ).rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3c5d6525-2744-450d-8998-180df31706a6",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(2181372, 8)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS       ID REF ALT QUAL FILTER  \\\n",
       "0     1  69134  2205837   A   G    .      .   \n",
       "1     1  69581  2252161   C   G    .      .   \n",
       "2     1  69682  2396347   G   A    .      .   \n",
       "3     1  69769  2288999   T   C    .      .   \n",
       "4     1  69995  2351346   G   C    .      .   \n",
       "\n",
       "                                                INFO  \n",
       "0  ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "1  ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "2  ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "3  ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...  \n",
       "4  ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Importing the ClinVar VCF file (as of 2023-06-26)\n",
    "# Source: https://ftp.ncbi.nlm.nih.gov/pub/clinvar/\n",
    "clinvar = read_vcf(path=cwd+\"/ClinVar_Data/clinvar_20230626.vcf\")\n",
    "display(clinvar.shape)\n",
    "display(clinvar.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "022b16e3-25b3-4cdb-8337-e03d8b1b7b61",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:C0950123;CLNDN=Inborn_genetic_diseases;CLNHGVS=NC_000001.10:g.69134A>G;CLNREVSTAT=criteria_provided,_single_submitter;CLNSIG=Likely_benign;CLNVC=single_nucleotide_variant;CLNVCSO=SO:0001483;GENEINFO=OR4F5:79501;MC=SO:0001583|missense_variant;ORIGIN=1'"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar[\"INFO\"][0]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f7bd888-a71d-46bd-ab63-32efc3fce15f",
   "metadata": {},
   "source": [
    "## Adding new columns based on ClinVar's \"INFO\" column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "710b2e25-fec7-425f-ab10-16685c84930f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done with ALLELEID\n",
      "Done with CLNDISDB\n",
      "Done with CLNDN\n",
      "Done with CLNHGVS\n",
      "Done with CLNREVSTAT\n",
      "Done with CLNSIG\n",
      "Done with CLNVC\n",
      "Done with CLNVCSO\n",
      "Done with GENEINFO\n",
      "Done with MC\n",
      "Done with ORIGIN\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>CLNDISDB</th>\n",
       "      <th>...</th>\n",
       "      <th>CLNHGVS</th>\n",
       "      <th>CLNREVSTAT</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>69134</td>\n",
       "      <td>2205837</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>2193183</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000001.10:g.69134A&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Likely_benign</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>69581</td>\n",
       "      <td>2252161</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>2238986</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000001.10:g.69581C&gt;G</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>69682</td>\n",
       "      <td>2396347</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>2386655</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000001.10:g.69682G&gt;A</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>69769</td>\n",
       "      <td>2288999</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>2278803</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000001.10:g.69769T&gt;C</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>69995</td>\n",
       "      <td>2351346</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...</td>\n",
       "      <td>2333177</td>\n",
       "      <td>MeSH:D030342,MedGen:C0950123</td>\n",
       "      <td>...</td>\n",
       "      <td>NC_000001.10:g.69995G&gt;C</td>\n",
       "      <td>criteria_provided,_single_submitter</td>\n",
       "      <td>Uncertain_significance</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>OR4F5:79501</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>79501</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  CHROM    POS       ID REF ALT QUAL FILTER  \\\n",
       "0     1  69134  2205837   A   G    .      .   \n",
       "1     1  69581  2252161   C   G    .      .   \n",
       "2     1  69682  2396347   G   A    .      .   \n",
       "3     1  69769  2288999   T   C    .      .   \n",
       "4     1  69995  2351346   G   C    .      .   \n",
       "\n",
       "                                                INFO ALLELEID  \\\n",
       "0  ALLELEID=2193183;CLNDISDB=MeSH:D030342,MedGen:...  2193183   \n",
       "1  ALLELEID=2238986;CLNDISDB=MeSH:D030342,MedGen:...  2238986   \n",
       "2  ALLELEID=2386655;CLNDISDB=MeSH:D030342,MedGen:...  2386655   \n",
       "3  ALLELEID=2278803;CLNDISDB=MeSH:D030342,MedGen:...  2278803   \n",
       "4  ALLELEID=2333177;CLNDISDB=MeSH:D030342,MedGen:...  2333177   \n",
       "\n",
       "                       CLNDISDB  ...                  CLNHGVS  \\\n",
       "0  MeSH:D030342,MedGen:C0950123  ...  NC_000001.10:g.69134A>G   \n",
       "1  MeSH:D030342,MedGen:C0950123  ...  NC_000001.10:g.69581C>G   \n",
       "2  MeSH:D030342,MedGen:C0950123  ...  NC_000001.10:g.69682G>A   \n",
       "3  MeSH:D030342,MedGen:C0950123  ...  NC_000001.10:g.69769T>C   \n",
       "4  MeSH:D030342,MedGen:C0950123  ...  NC_000001.10:g.69995G>C   \n",
       "\n",
       "                            CLNREVSTAT                  CLNSIG  \\\n",
       "0  criteria_provided,_single_submitter           Likely_benign   \n",
       "1  criteria_provided,_single_submitter  Uncertain_significance   \n",
       "2  criteria_provided,_single_submitter  Uncertain_significance   \n",
       "3  criteria_provided,_single_submitter  Uncertain_significance   \n",
       "4  criteria_provided,_single_submitter  Uncertain_significance   \n",
       "\n",
       "                       CLNVC     CLNVCSO     GENEINFO   Gene Gene_ID  \\\n",
       "0  single_nucleotide_variant  SO:0001483  OR4F5:79501  OR4F5   79501   \n",
       "1  single_nucleotide_variant  SO:0001483  OR4F5:79501  OR4F5   79501   \n",
       "2  single_nucleotide_variant  SO:0001483  OR4F5:79501  OR4F5   79501   \n",
       "3  single_nucleotide_variant  SO:0001483  OR4F5:79501  OR4F5   79501   \n",
       "4  single_nucleotide_variant  SO:0001483  OR4F5:79501  OR4F5   79501   \n",
       "\n",
       "                            MC ORIGIN  \n",
       "0  SO:0001583|missense_variant    NaN  \n",
       "1  SO:0001583|missense_variant    NaN  \n",
       "2  SO:0001583|missense_variant    NaN  \n",
       "3  SO:0001583|missense_variant    NaN  \n",
       "4  SO:0001583|missense_variant    NaN  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# ClinVar Allele ID\n",
    "pattern = r\"ALLELEID=(.*?);\"\n",
    "clinvar[\"ALLELEID\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with ALLELEID\")\n",
    "\n",
    "# tag-value pairs of disease database name and identifier\n",
    "pattern = r\"CLNDISDB=(.*?);\"\n",
    "clinvar[\"CLNDISDB\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNDISDB\")\n",
    "\n",
    "# ClinVar's preferred disease name for the concept specified by disease identifiers in CLNDISDB\n",
    "pattern = r\"CLNDN=(.*?);\"\n",
    "clinvar[\"CLNDN\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNDN\")\n",
    "\n",
    "# top-level (primary assembly, alt, or patch) HGVS expression\n",
    "pattern = r\"CLNHGVS=(.*?);\"\n",
    "clinvar[\"CLNHGVS\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNHGVS\")\n",
    "\n",
    "# ClinVar's review status for the Variation ID\n",
    "pattern = r\"CLNREVSTAT=(.*?);\"\n",
    "clinvar[\"CLNREVSTAT\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNREVSTAT\")\n",
    "\n",
    "# clinical significance for this single variant; multiple values are separated by a vertical bar\n",
    "pattern = r\"CLNSIG=(.*?);\"\n",
    "clinvar[\"CLNSIG\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNSIG\")\n",
    "\n",
    "# variant type\n",
    "pattern = r\"CLNVC=(.*?);\"\n",
    "clinvar[\"CLNVC\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNVC\")\n",
    "\n",
    "# Sequence Ontology ID for the variant type (www.sequenceontology.org)\n",
    "pattern = r\"CLNVCSO=(.*?);\"\n",
    "clinvar[\"CLNVCSO\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with CLNVCSO\")\n",
    "\n",
    "# gene(s) for the variant reported as gene symbol:NCBI GeneID. The gene symbol and ID are delimited by a colon and each pair is delimited by a vertical bar.\n",
    "pattern = r\"GENEINFO=(.*?);\"\n",
    "clinvar[\"GENEINFO\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "clinvar[[\"Gene\", \"Gene_ID\"]] = clinvar[\"GENEINFO\"].str.split(\":\", 1, expand=True)\n",
    "print(\"Done with GENEINFO\")\n",
    "\n",
    "# comma separated list of molecular consequence in the form of Sequence Ontology ID|molecular_consequence\n",
    "pattern = r\"MC=(.*?);\"\n",
    "clinvar[\"MC\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with MC\")\n",
    "\n",
    "# Allele origin reported to ClinVar\n",
    "# 0 = unknown\n",
    "# 1 = germline\n",
    "# 2 = somatic \n",
    "# 4 = inherited\n",
    "# 8 = paternal \n",
    "# 16 = maternal \n",
    "# 32 = de-novo\n",
    "# 64 = biparental\n",
    "# 128 = uniparental \n",
    "# 256 = not-tested\n",
    "# 512 = tested-inconclusive\n",
    "# 1073741824 = other\n",
    "pattern = r\"ORIGIN=(.*?);\"\n",
    "clinvar[\"ORIGIN\"] = clinvar[\"INFO\"].str.extract(pattern)\n",
    "print(\"Done with ORIGIN\")\n",
    "\n",
    "display(clinvar.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6e556781-0115-45c4-8090-2a19ca20d151",
   "metadata": {},
   "source": [
    "## Filtering for BLB and PLP for any SNV reported to ClinVar after 2020-11-21 (MutScore training cutoff)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5219a33b-8517-4fed-b0e1-9bcc29c297c1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO',\n",
       "       'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', 'CLNREVSTAT', 'CLNSIG',\n",
       "       'CLNVC', 'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "067928cb-6b73-40d4-9763-be5b2738d306",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['single_nucleotide_variant', 'Microsatellite', 'Indel', 'Deletion',\n",
       "       'Duplication', 'Insertion', 'Variation', 'Inversion'], dtype=object)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar[\"CLNVC\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "774c740c-30d0-419f-96dd-0fbf26fcedef",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Likely_benign', 'Uncertain_significance', 'Benign',\n",
       "       'Conflicting_interpretations_of_pathogenicity', 'Pathogenic',\n",
       "       'Likely_pathogenic', 'Benign/Likely_benign', 'not_provided',\n",
       "       'Pathogenic/Likely_pathogenic', nan, 'risk_factor', 'Affects',\n",
       "       'association', 'Benign|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|other',\n",
       "       'drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|association',\n",
       "       'Uncertain_risk_allele', 'other',\n",
       "       'Uncertain_significance|risk_factor',\n",
       "       'Likely_pathogenic|risk_factor', 'Likely_benign|association',\n",
       "       'Likely_risk_allele', 'Pathogenic/Likely_pathogenic|other',\n",
       "       'Pathogenic|other',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance|other',\n",
       "       'Pathogenic/Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'protective', 'Pathogenic|risk_factor',\n",
       "       'Pathogenic/Likely_pathogenic|risk_factor',\n",
       "       'Benign/Likely_benign|risk_factor',\n",
       "       'Uncertain_significance/Uncertain_risk_allele',\n",
       "       'Pathogenic|association|protective', 'protective|risk_factor',\n",
       "       'Benign/Likely_benign|other|risk_factor',\n",
       "       'Benign/Likely_benign|association', 'Benign|association',\n",
       "       'Affects|association', 'Pathogenic|protective',\n",
       "       'Benign|risk_factor',\n",
       "       'Conflicting_interpretations_of_pathogenicity|drug_response|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|drug_response',\n",
       "       'Benign|drug_response', 'Likely_pathogenic|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|protective',\n",
       "       'Pathogenic/Likely_pathogenic|drug_response',\n",
       "       'Benign/Likely_benign|other',\n",
       "       'Likely_pathogenic/Likely_risk_allele',\n",
       "       'Uncertain_risk_allele|protective', 'association_not_found',\n",
       "       'Uncertain_significance|association', 'Likely_benign|other',\n",
       "       'Uncertain_significance|other',\n",
       "       'Conflicting_interpretations_of_pathogenicity|association|risk_factor',\n",
       "       'Pathogenic|association',\n",
       "       'Conflicting_interpretations_of_pathogenicity|other|risk_factor',\n",
       "       'drug_response|risk_factor',\n",
       "       'Uncertain_significance|drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|risk_factor',\n",
       "       'Pathogenic/Likely_pathogenic/Likely_risk_allele',\n",
       "       'Pathogenic/Likely_risk_allele', 'Likely_pathogenic|drug_response',\n",
       "       'Pathogenic|drug_response',\n",
       "       'Conflicting_interpretations_of_pathogenicity|Affects',\n",
       "       'association|drug_response|risk_factor', 'Pathogenic|Affects',\n",
       "       'Affects|risk_factor', 'Pathogenic|drug_response|other',\n",
       "       'confers_sensitivity', 'Likely_pathogenic|Affects',\n",
       "       'Uncertain_risk_allele|risk_factor',\n",
       "       'Likely_benign|drug_response|other',\n",
       "       'Benign/Likely_benign|drug_response',\n",
       "       'Benign/Likely_benign|drug_response|other', 'drug_response|other',\n",
       "       'association|drug_response', 'Pathogenic|confers_sensitivity',\n",
       "       'Likely_pathogenic,_low_penetrance', 'association|risk_factor',\n",
       "       'Benign|confers_sensitivity',\n",
       "       'Likely_pathogenic/Pathogenic,_low_penetrance',\n",
       "       'Likely_benign|risk_factor', 'Likely_pathogenic|association'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar[\"CLNSIG\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "0963ff80-4f9d-4bd1-a8c0-f7eeaf74d0ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Clinvar unfiltered: (2181372, 21)\n",
      "Filtered for SNVs: (1991708, 21)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\TooFastDan\\AppData\\Local\\Temp\\ipykernel_18304\\1611858059.py:7: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  clinvar_filt[\"ID\"] = clinvar_filt[\"ID\"].astype(int)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Filtered for dates after 2020-11-21: (1291946, 21)\n",
      "Filtered for pathogenicity annotations: (573694, 21)\n"
     ]
    }
   ],
   "source": [
    "# Filtering for SNVs\n",
    "clinvar_filt = clinvar[clinvar[\"CLNVC\"]==\"single_nucleotide_variant\"]\n",
    "print(\"Clinvar unfiltered:\", clinvar.shape)\n",
    "print(\"Filtered for SNVs:\", clinvar_filt.shape)\n",
    "\n",
    "# Filtering out dates AFTER 2020-11-21\n",
    "clinvar_filt[\"ID\"] = clinvar_filt[\"ID\"].astype(int)\n",
    "clinvar_filt = clinvar_filt.sort_values(by=\"ID\", ascending=False)\n",
    "clinvar_filt = clinvar_filt[clinvar_filt[\"ID\"] > 986732]  #if you look up ID 986732 in clinvar, this is the first SNV reported to clinvar after 2020-11-21\n",
    "print(\"Filtered for dates after 2020-11-21:\", clinvar_filt.shape)\n",
    "\n",
    "# Filtering for variants with PLP and BLB annotations only \n",
    "pathogenecity = [\"Benign\", \"Likely_benign\", \"Benign/Likely_benign\", \"Pathogenic\", \"Likely_pathogenic\", \"Pathogenic/Likely_pathogenic\"]\n",
    "clinvar_filt = clinvar_filt[clinvar_filt[\"CLNSIG\"].isin(pathogenecity)]\n",
    "print(\"Filtered for pathogenicity annotations:\", clinvar_filt.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "31181bf7-3685-4a20-b920-cba124674441",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(573694, 24)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>PATHOGENICITY</th>\n",
       "      <th>y_test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2153503</th>\n",
       "      <td>chrX</td>\n",
       "      <td>107869448</td>\n",
       "      <td>107869448</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=35936;CLNDISDB=MONDO:MONDO:0010520,Me...</td>\n",
       "      <td>2506495</td>\n",
       "      <td>35936</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL4A5:1287</td>\n",
       "      <td>COL4A5</td>\n",
       "      <td>1287</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>1</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2153970</th>\n",
       "      <td>chrX</td>\n",
       "      <td>107930842</td>\n",
       "      <td>107930842</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671452;CLNDISDB=MONDO:MONDO:0010520,...</td>\n",
       "      <td>2506494</td>\n",
       "      <td>2671452</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL4A5:1287</td>\n",
       "      <td>COL4A5</td>\n",
       "      <td>1287</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1081798</th>\n",
       "      <td>chr10</td>\n",
       "      <td>16870957</td>\n",
       "      <td>16870957</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671451;CLNDISDB=MONDO:MONDO:0030042,...</td>\n",
       "      <td>2506493</td>\n",
       "      <td>2671451</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>CUBN:8029</td>\n",
       "      <td>CUBN</td>\n",
       "      <td>8029</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382434</th>\n",
       "      <td>chr2</td>\n",
       "      <td>228124576</td>\n",
       "      <td>228124576</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671449;CLNDISDB=MONDO:MONDO:0008762,...</td>\n",
       "      <td>2506491</td>\n",
       "      <td>2671449</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL4A3:1285|MFF-DT:654841</td>\n",
       "      <td>COL4A3</td>\n",
       "      <td>1285|MFF-DT:654841</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>382921</th>\n",
       "      <td>chr2</td>\n",
       "      <td>228145714</td>\n",
       "      <td>228145714</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671448;CLNDISDB=MONDO:MONDO:0008762,...</td>\n",
       "      <td>2506490</td>\n",
       "      <td>2671448</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL4A3:1285|MFF-DT:654841</td>\n",
       "      <td>COL4A3</td>\n",
       "      <td>1285|MFF-DT:654841</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM      START        END REF ALT QUAL FILTER  \\\n",
       "2153503   chrX  107869448  107869448   G   A    .      .   \n",
       "2153970   chrX  107930842  107930842   C   A    .      .   \n",
       "1081798  chr10   16870957   16870957   G   T    .      .   \n",
       "382434    chr2  228124576  228124576   G   A    .      .   \n",
       "382921    chr2  228145714  228145714   G   A    .      .   \n",
       "\n",
       "                                                      INFO       ID ALLELEID  \\\n",
       "2153503  ALLELEID=35936;CLNDISDB=MONDO:MONDO:0010520,Me...  2506495    35936   \n",
       "2153970  ALLELEID=2671452;CLNDISDB=MONDO:MONDO:0010520,...  2506494  2671452   \n",
       "1081798  ALLELEID=2671451;CLNDISDB=MONDO:MONDO:0030042,...  2506493  2671451   \n",
       "382434   ALLELEID=2671449;CLNDISDB=MONDO:MONDO:0008762,...  2506491  2671449   \n",
       "382921   ALLELEID=2671448;CLNDISDB=MONDO:MONDO:0008762,...  2506490  2671448   \n",
       "\n",
       "         ...                      CLNVC     CLNVCSO  \\\n",
       "2153503  ...  single_nucleotide_variant  SO:0001483   \n",
       "2153970  ...  single_nucleotide_variant  SO:0001483   \n",
       "1081798  ...  single_nucleotide_variant  SO:0001483   \n",
       "382434   ...  single_nucleotide_variant  SO:0001483   \n",
       "382921   ...  single_nucleotide_variant  SO:0001483   \n",
       "\n",
       "                          GENEINFO    Gene             Gene_ID  \\\n",
       "2153503                COL4A5:1287  COL4A5                1287   \n",
       "2153970                COL4A5:1287  COL4A5                1287   \n",
       "1081798                  CUBN:8029    CUBN                8029   \n",
       "382434   COL4A3:1285|MFF-DT:654841  COL4A3  1285|MFF-DT:654841   \n",
       "382921   COL4A3:1285|MFF-DT:654841  COL4A3  1285|MFF-DT:654841   \n",
       "\n",
       "                                  MC ORIGIN             CLNSIG PATHOGENICITY  \\\n",
       "2153503  SO:0001583|missense_variant      1  Likely_pathogenic           PLP   \n",
       "2153970          SO:0001587|nonsense    NaN  Likely_pathogenic           PLP   \n",
       "1081798          SO:0001587|nonsense    NaN  Likely_pathogenic           PLP   \n",
       "382434   SO:0001583|missense_variant    NaN         Pathogenic           PLP   \n",
       "382921   SO:0001583|missense_variant    NaN  Likely_pathogenic           PLP   \n",
       "\n",
       "        y_test  \n",
       "2153503      1  \n",
       "2153970      1  \n",
       "1081798      1  \n",
       "382434       1  \n",
       "382921       1  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Cleaning and mapping data \n",
    "\n",
    "# Mapping and binarizing pathogenicity\n",
    "clinvar_filt[\"PATHOGENICITY\"] = clinvar_filt[\"CLNSIG\"].map(\n",
    "    {\"Benign\":\"BLB\", \"Likely_benign\":\"BLB\", \"Benign/Likely_benign\":\"BLB\", \"Pathogenic\":\"PLP\", \"Likely_pathogenic\":\"PLP\", \"Pathogenic/Likely_pathogenic\":\"PLP\"})\n",
    "clinvar_filt[\"y_test\"] = clinvar_filt[\"PATHOGENICITY\"].map({\"PLP\": 1, \"BLB\": 0})\n",
    "\n",
    "# Renaming chromosomes for ANNOVAR input\n",
    "chromosomes = {'1':'chr1', '2':'chr2', '3':'chr3', '4':'chr4', '5':'chr5', \n",
    "                '6':'chr6', '7':'chr7', '8':'chr8', '9':'chr9', '10':'chr10', \n",
    "                '11':'chr11', '12':'chr12', '13':'chr13', '14':'chr14', '15':'chr15', \n",
    "                '16':'chr16', '17':'chr17', '18':'chr18', '19':'chr19', '20':'chr20', \n",
    "                '21':'chr21', '22':'chr22', 'X':'chrX', 'Y':'chrY', 'MT':'M'}\n",
    "clinvar_filt[\"CHROM\"] = clinvar_filt[\"CHROM\"].map(chromosomes)\n",
    "\n",
    "# Making the end position the same name as the start position for SNVs and reordering/renaming columns for ANNOVAR input\n",
    "clinvar_filt[\"END\"] = clinvar_filt[\"POS\"]\n",
    "clinvar_filt = clinvar_filt[['CHROM', 'POS', 'END', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'ID', 'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', 'CLNREVSTAT', 'CLNVC', 'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN', 'CLNSIG', 'PATHOGENICITY', 'y_test']]\n",
    "clinvar_filt.columns = ['CHROM', 'START', 'END', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'ID', 'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', 'CLNREVSTAT', 'CLNVC', 'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN', 'CLNSIG', 'PATHOGENICITY', 'y_test']\n",
    "\n",
    "# Displaying the filtered df\n",
    "display(clinvar_filt.shape)\n",
    "display(clinvar_filt.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "797e0e44-0b37-4799-95c5-e3ce4ab8c248",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CHROM', 'START', 'END', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'ID',\n",
       "       'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', 'CLNREVSTAT', 'CLNVC',\n",
       "       'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN', 'CLNSIG',\n",
       "       'PATHOGENICITY', 'y_test'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_filt.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e534e4bb-d69c-4bea-89ca-4c97755a4ce1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Likely_benign                   433231\n",
       "Benign                           87625\n",
       "Pathogenic                       24480\n",
       "Likely_pathogenic                22855\n",
       "Benign/Likely_benign              3691\n",
       "Pathogenic/Likely_pathogenic      1812\n",
       "Name: CLNSIG, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_filt[\"CLNSIG\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "0f3686e9-c4bb-43f1-bfa4-be3ddf92536e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BLB    524547\n",
       "PLP     49147\n",
       "Name: PATHOGENICITY, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_filt[\"PATHOGENICITY\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "151834cc-52a4-4037-9572-2ebe649bbf1e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CHROM            object\n",
       "START             int32\n",
       "END               int32\n",
       "REF              object\n",
       "ALT              object\n",
       "QUAL             object\n",
       "FILTER           object\n",
       "INFO             object\n",
       "ID                int32\n",
       "ALLELEID         object\n",
       "CLNDISDB         object\n",
       "CLNDN            object\n",
       "CLNHGVS          object\n",
       "CLNREVSTAT       object\n",
       "CLNVC            object\n",
       "CLNVCSO          object\n",
       "GENEINFO         object\n",
       "Gene             object\n",
       "Gene_ID          object\n",
       "MC               object\n",
       "ORIGIN           object\n",
       "CLNSIG           object\n",
       "PATHOGENICITY    object\n",
       "y_test            int64\n",
       "dtype: object"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinvar_filt.dtypes"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ab3247a-d9c0-47cd-8b1f-78ed85eea209",
   "metadata": {},
   "source": [
    "## Exporting as a txt file with the extention \"avinput\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "ae283a05-e550-4430-8363-abb0bd3bb4ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "#clinvar_filt.to_csv(cwd+'/annovar_files/clinvar_filt_20201128-20230626.avinput', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3153800e-7bd5-42a7-854a-3d5c616da66b",
   "metadata": {},
   "source": [
    "### Optional filtering using only RetNet genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "9880a10c-6e57-4fe2-9baf-0d0b34cd69cc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62696, 24)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>PATHOGENICITY</th>\n",
       "      <th>y_test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1107483</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...</td>\n",
       "      <td>2506469</td>\n",
       "      <td>2671427</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>HK1:3098</td>\n",
       "      <td>HK1</td>\n",
       "      <td>3098</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1989824</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...</td>\n",
       "      <td>2506465</td>\n",
       "      <td>2671423</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>JAG1:182</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>182</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>749118</th>\n",
       "      <td>chr6</td>\n",
       "      <td>65300662</td>\n",
       "      <td>65300662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...</td>\n",
       "      <td>2506396</td>\n",
       "      <td>2671350</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>EYS:346007</td>\n",
       "      <td>EYS</td>\n",
       "      <td>346007</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>537620</th>\n",
       "      <td>chr4</td>\n",
       "      <td>39255536</td>\n",
       "      <td>39255536</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...</td>\n",
       "      <td>2506391</td>\n",
       "      <td>2671345</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>WDR19:57728</td>\n",
       "      <td>WDR19</td>\n",
       "      <td>57728</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1320901</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...</td>\n",
       "      <td>2506270</td>\n",
       "      <td>2671226</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL2A1:1280</td>\n",
       "      <td>COL2A1</td>\n",
       "      <td>1280</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM     START       END REF ALT QUAL FILTER  \\\n",
       "1107483  chr10  71142347  71142347   C   A    .      .   \n",
       "1989824  chr20  10654126  10654126   A   C    .      .   \n",
       "749118    chr6  65300662  65300662   T   A    .      .   \n",
       "537620    chr4  39255536  39255536   C   T    .      .   \n",
       "1320901  chr12  48380959  48380959   C   T    .      .   \n",
       "\n",
       "                                                      INFO       ID ALLELEID  \\\n",
       "1107483  ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...  2506469  2671427   \n",
       "1989824  ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...  2506465  2671423   \n",
       "749118   ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...  2506396  2671350   \n",
       "537620   ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...  2506391  2671345   \n",
       "1320901  ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...  2506270  2671226   \n",
       "\n",
       "         ...                      CLNVC     CLNVCSO     GENEINFO    Gene  \\\n",
       "1107483  ...  single_nucleotide_variant  SO:0001483     HK1:3098     HK1   \n",
       "1989824  ...  single_nucleotide_variant  SO:0001483     JAG1:182    JAG1   \n",
       "749118   ...  single_nucleotide_variant  SO:0001483   EYS:346007     EYS   \n",
       "537620   ...  single_nucleotide_variant  SO:0001483  WDR19:57728   WDR19   \n",
       "1320901  ...  single_nucleotide_variant  SO:0001483  COL2A1:1280  COL2A1   \n",
       "\n",
       "        Gene_ID                           MC ORIGIN             CLNSIG  \\\n",
       "1107483    3098  SO:0001583|missense_variant    NaN  Likely_pathogenic   \n",
       "1989824     182  SO:0001583|missense_variant    NaN  Likely_pathogenic   \n",
       "749118   346007          SO:0001587|nonsense    NaN         Pathogenic   \n",
       "537620    57728          SO:0001587|nonsense    NaN  Likely_pathogenic   \n",
       "1320901    1280  SO:0001583|missense_variant    NaN  Likely_pathogenic   \n",
       "\n",
       "        PATHOGENICITY y_test  \n",
       "1107483           PLP      1  \n",
       "1989824           PLP      1  \n",
       "749118            PLP      1  \n",
       "537620            PLP      1  \n",
       "1320901           PLP      1  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Importing retnet genes\n",
    "retnet = pd.read_excel(cwd+\"/RetNet/RetNet_AD-AR_FINALIZED.xlsx\", sheet_name=\"gene_info\")\n",
    "retnet_genes = retnet[\"gene\"]\n",
    "\n",
    "# Filtering filtered ClinVar variants to only include RetNet Genes\n",
    "clinvar_filt_retnet = clinvar_filt[clinvar_filt[\"Gene\"].isin(retnet_genes)]\n",
    "display(clinvar_filt_retnet.shape)\n",
    "display(clinvar_filt_retnet.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d3bc1b42-db8c-45d4-a8a5-aadb4eff8b1a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "265"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(clinvar_filt_retnet[\"Gene\"].unique())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "100aa058-4add-476b-8b22-b44e44d2ffdc",
   "metadata": {},
   "source": [
    "## Exporting filtered RetNet SNVs as a txt file with the extention \"avinput\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e51b0ec9-8c14-4369-a71c-2d2c4b70cfaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "clinvar_filt_retnet.to_csv(cwd+'/annovar_files/clinvar_filt_retnet.avinput', sep=\"\\t\", index=False, header=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a27f94c3-7def-4be6-8488-27427b737885",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc78a4bc-b93b-4c18-af0e-2580521a1e34",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05575e56-2f7b-460c-b12c-8bc7a63dc834",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e542eeb-a657-42d1-a9a6-c07a94bc177d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3f34d4c-c7ba-425c-b23b-4fdd0bdb2288",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "97fe70a2-f371-4797-9950-f7cd5d86dd1b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b5adb521-3851-4e13-a6c6-7f761b8990e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2cbfc7f-0315-4ef2-a3e4-31b4fc7ac31e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e9376dc-c22e-4acf-9d2c-34523721d6f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "590e69f5-b6f8-4166-b3ee-6c317258479b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "61aca353-0406-4c9f-a615-50e2b473a878",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb8e645d-7d7d-4708-8243-91cdc8300ef4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
