{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "eab3b847-2129-49f7-b488-a617db5479b4",
   "metadata": {},
   "source": [
    "# **Classifier Thresholds for Likely Benign and Likely Pathogenic SNVs** \n",
    "\n",
    "### Author: Daniel Brock\n",
    "### Date: 7/15/2023\n",
    "### Purpose: to define the upper and lower threshold value for each model (percent at while predictor is 95% accurate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "c45a4923-b453-4b8b-a867-0452e3fd5a31",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\TooFastDan\\OneDrive - Baylor College of Medicine\\BCM\\Projects\\Autosomal Dominant Predictor of IRDs\\manuscript\\GitHub\n"
     ]
    }
   ],
   "source": [
    "# Importing required packages\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "# Setting working directory\n",
    "cwd = os.getcwd()\n",
    "print(cwd)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ac32d1b9-9b75-4eff-b9c2-3b5e457bf194",
   "metadata": {},
   "source": [
    "## **Defining Functions used to calculate upper and lower threshold values for classifiers**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "045b8b60-9e90-49fa-8930-ab43b5fbd3d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "def likely_pathogenic(min_score, max_score, df, classifier, iterations=1001):\n",
    "    \"\"\"\n",
    "    Function to calculate the upper threshold of a classifier model, where 95% of variants from the training ClinVar set are correctly classified as PLP.\n",
    "    Inputs:\n",
    "    - min_score: minimum score of a classifier\n",
    "    - max_score: maximum score of a classifier\n",
    "    - df: dataframe containing variants, classifier scores, and pathogenicity classifications\n",
    "    - classifier: name of the classifier model\n",
    "    Outputs:\n",
    "    - scores dataframe: percent_PLP and percent_BLB for each scoring option\n",
    "    - LP_threshold: threshold for a variant being likely pathogenic (LP)\n",
    "    \"\"\"\n",
    "    \n",
    "    # Creating an array of scores in the thousandths decimal place (used by classifiers)\n",
    "    scores = np.linspace(min_score, max_score, num=iterations)\n",
    "    scores = np.around(scores, 3)\n",
    "    \n",
    "    # Looping through the ClinVar variants data to get percent PLP and percent BLB for each possible classifier score\n",
    "    df2 = df[df[classifier].notna()]\n",
    "    scores_df_list = []\n",
    "    for score in scores:\n",
    "        s_df = df2[df2[classifier] >= score]\n",
    "        plps = s_df[s_df[\"PATHOGENICITY\"]==\"PLP\"].shape[0] #number of PLP variants classified by clinvar\n",
    "        blbs = s_df[s_df['PATHOGENICITY']==\"BLB\"].shape[0] #number of BLB variants classified by clinvar\n",
    "        pb = pd.DataFrame({'PLP': [plps], 'BLB': [blbs]})\n",
    "        pb[\"percent_PLP\"] = pb[\"PLP\"] / (pb[\"PLP\"] + pb[\"BLB\"]) * 100\n",
    "        pb[\"percent_BLB\"] = pb[\"BLB\"] / (pb[\"PLP\"] + pb[\"BLB\"]) * 100\n",
    "        pb[classifier] = score\n",
    "        scores_df_list.append(pb)\n",
    "    scores_df = pd.concat(scores_df_list, axis=0)\n",
    "    \n",
    "    # Returning scores df and likely pathogenic threshold\n",
    "    try:\n",
    "        lp = min(scores_df[scores_df[\"percent_PLP\"] > 95][classifier])\n",
    "    except:\n",
    "        lp = np.nan\n",
    "        print(\"Had Issues\")\n",
    "    return scores_df, lp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "0e31595d-4fec-482c-a909-cea8c46efaed",
   "metadata": {},
   "outputs": [],
   "source": [
    "def likely_benign(min_score, max_score, df, classifier, iterations=1001):\n",
    "    \"\"\"\n",
    "    Function to calculate the lower threshold of a classifier model, where 95% of variants from the training ClinVar set are correctly classified as BLB.\n",
    "    Inputs:\n",
    "    - min_score: minimum score of a classifier\n",
    "    - max_score: maximum score of a classifier\n",
    "    - df: dataframe containing variants, classifier scores, and pathogenicity classifications\n",
    "    - classifier: name of the classifier model\n",
    "    Outputs:\n",
    "    - scores dataframe: percent_PLP and percent_BLB for each scoring option\n",
    "    - LP_threshold: threshold for a variant being likely benign (LB)\n",
    "    \"\"\"\n",
    "    \n",
    "    # Creating an array of scores in the thousandths decimal place (used by classifiers)\n",
    "    scores = np.linspace(min_score, max_score, num=iterations)\n",
    "    scores = np.around(scores, 3)\n",
    "    \n",
    "    # Looping through the ClinVar variants data to get percent PLP and percent BLB for each possible classifier score\n",
    "    df2 = df[df[classifier].notna()]\n",
    "    scores_df_list = []\n",
    "    for score in scores:\n",
    "        s_df = df2[df2[classifier] <= score]\n",
    "        plps = s_df[s_df[\"PATHOGENICITY\"]==\"PLP\"].shape[0] #number of PLP variants classified by clinvar\n",
    "        blbs = s_df[s_df['PATHOGENICITY']==\"BLB\"].shape[0] #number of BLB variants classified by clinvar\n",
    "        pb = pd.DataFrame({'PLP': [plps], 'BLB': [blbs]})\n",
    "        pb[\"percent_PLP\"] = pb[\"PLP\"] / (pb[\"PLP\"] + pb[\"BLB\"]) * 100\n",
    "        pb[\"percent_BLB\"] = pb[\"BLB\"] / (pb[\"PLP\"] + pb[\"BLB\"]) * 100\n",
    "        pb[classifier] = score\n",
    "        scores_df_list.append(pb)\n",
    "    scores_df = pd.concat(scores_df_list, axis=0)\n",
    "    \n",
    "    # Returning scores df and likely pathogenic threshold\n",
    "    lp = max(scores_df[scores_df[\"percent_BLB\"] > 95][classifier])\n",
    "    return scores_df, lp"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "951a42e7-3670-428c-8bc6-71b9a7e42337",
   "metadata": {},
   "source": [
    "# **Getting Threshold Values for ALL untrained ClinVar RetNet Genes**\n",
    "\n",
    "### Source: Clinvar VCF File: https://www.ncbi.nlm.nih.gov/clinvar/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "2b7bff0f-54da-4231-94d4-c8c021134236",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3330, 154)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>OMIM Phenotypes</th>\n",
       "      <th>How Identified;_x000D_\\nComments</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>pLI</th>\n",
       "      <th>exac_pLI</th>\n",
       "      <th>AR</th>\n",
       "      <th>AD</th>\n",
       "      <th>AD GOF</th>\n",
       "      <th>AD Haploinsuffiency</th>\n",
       "      <th>Complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...</td>\n",
       "      <td>2506469</td>\n",
       "      <td>2671427</td>\n",
       "      <td>...</td>\n",
       "      <td>Retinitis pigmentosa 79, 617460 (3), Autosomal...</td>\n",
       "      <td>linkage mapping, whole-exome sequencing; a mis...</td>\n",
       "      <td>10</td>\n",
       "      <td>9.145900e-01</td>\n",
       "      <td>0.555730</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...</td>\n",
       "      <td>2506465</td>\n",
       "      <td>2671423</td>\n",
       "      <td>...</td>\n",
       "      <td>?Deafness, congenital heart defects, and poste...</td>\n",
       "      <td>deletion mapping, candidate gene; multiple aff...</td>\n",
       "      <td>20</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...</td>\n",
       "      <td>2506270</td>\n",
       "      <td>2671226</td>\n",
       "      <td>...</td>\n",
       "      <td>?Vitreoretinopathy with phalangeal epiphyseal ...</td>\n",
       "      <td>linkage mapping, candidate gene; mutations in ...</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr20</td>\n",
       "      <td>3897663</td>\n",
       "      <td>3897663</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALLELEID=2671068;CLNDISDB=MONDO:MONDO:0009319,...</td>\n",
       "      <td>2506108</td>\n",
       "      <td>2671068</td>\n",
       "      <td>...</td>\n",
       "      <td>HARP syndrome, 607236 (3), Autosomal recessive...</td>\n",
       "      <td>homozygosity mapping, candidate gene; symptoms...</td>\n",
       "      <td>20</td>\n",
       "      <td>7.403400e-07</td>\n",
       "      <td>0.010792</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48378867</td>\n",
       "      <td>48378867</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ALLELEID=2670853;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
       "      <td>2505889</td>\n",
       "      <td>2670853</td>\n",
       "      <td>...</td>\n",
       "      <td>?Vitreoretinopathy with phalangeal epiphyseal ...</td>\n",
       "      <td>linkage mapping, candidate gene; mutations in ...</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 154 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Ref Alt  QUAL  FILTER  \\\n",
       "0  chr10  71142347  71142347   C   A   NaN     NaN   \n",
       "1  chr20  10654126  10654126   A   C   NaN     NaN   \n",
       "2  chr12  48380959  48380959   C   T   NaN     NaN   \n",
       "3  chr20   3897663   3897663   T   A   NaN     NaN   \n",
       "4  chr12  48378867  48378867   C   T   NaN     NaN   \n",
       "\n",
       "                                                INFO       ID  ALLELEID  ...  \\\n",
       "0  ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...  2506469   2671427  ...   \n",
       "1  ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...  2506465   2671423  ...   \n",
       "2  ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...  2506270   2671226  ...   \n",
       "3  ALLELEID=2671068;CLNDISDB=MONDO:MONDO:0009319,...  2506108   2671068  ...   \n",
       "4  ALLELEID=2670853;CLNDISDB=MedGen:CN517202;CLND...  2505889   2670853  ...   \n",
       "\n",
       "                                     OMIM Phenotypes  \\\n",
       "0  Retinitis pigmentosa 79, 617460 (3), Autosomal...   \n",
       "1  ?Deafness, congenital heart defects, and poste...   \n",
       "2  ?Vitreoretinopathy with phalangeal epiphyseal ...   \n",
       "3  HARP syndrome, 607236 (3), Autosomal recessive...   \n",
       "4  ?Vitreoretinopathy with phalangeal epiphyseal ...   \n",
       "\n",
       "                    How Identified;_x000D_\\nComments Chromosome           pLI  \\\n",
       "0  linkage mapping, whole-exome sequencing; a mis...         10  9.145900e-01   \n",
       "1  deletion mapping, candidate gene; multiple aff...         20  1.000000e+00   \n",
       "2  linkage mapping, candidate gene; mutations in ...         12  1.000000e+00   \n",
       "3  homozygosity mapping, candidate gene; symptoms...         20  7.403400e-07   \n",
       "4  linkage mapping, candidate gene; mutations in ...         12  1.000000e+00   \n",
       "\n",
       "   exac_pLI AR AD AD GOF AD Haploinsuffiency Complex  \n",
       "0  0.555730  1  1      1                   0     1.0  \n",
       "1  1.000000  0  1      0                   1     NaN  \n",
       "2  1.000000  0  1      0                   1     NaN  \n",
       "3  0.010792  1  0      0                   0     NaN  \n",
       "4  1.000000  0  1      0                   1     NaN  \n",
       "\n",
       "[5 rows x 154 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rc = pd.read_excel(cwd+\"/annovar_files/clinvar_filt_retnet_annotated_PLP-BLB_Nonsynonomous_Variants.xlsx\", na_values=\".\")\n",
    "display(rc.shape)\n",
    "display(rc.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "9c66ff5b-f67f-408d-9a54-f5cef070228f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Skip if done with preprocessed data\n",
    "\n",
    "# Importing annotated clinvar retnet variants\n",
    "#rc = pd.read_csv(cwd+\"/annotated_clinvar_filt_retnet_20230626.hg19_multianno.csv\", na_values=\".\")\n",
    "\n",
    "# Importing the clinvar retnet-filtered avinput file, which as the pathogenicity scores\n",
    "#clin_avi = pd.read_table(cwd+\"/clinvar_filt_retnet_20230626.avinput\", sep=\"\\t\",\n",
    "#                        names=['Chr', 'Start', 'End', 'Ref', 'Alt', 'QUAL', 'FILTER', 'INFO', 'ID', 'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', 'CLNREVSTAT', 'CLNVC', 'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN', 'CLNSIG', 'Pathogenicity', 'y_test'])\n",
    "#clin_avi = clin_avi[['Chr', 'Start', 'End', 'Ref', 'Alt', 'Pathogenicity', 'y_test']]\n",
    "#display(clin_avi.shape)\n",
    "#display(clin_avi.head())\n",
    "\n",
    "# Merging to get pathogenicity rankings\n",
    "#rc = pd.merge(rc, clin_avi, on=['Chr', 'Start', 'End', 'Ref', 'Alt'])\n",
    "\n",
    "# Filtering for nonsynonymous SNVs\n",
    "#rc = rc[rc[\"ExonicFunc.refGene\"]==\"nonsynonymous SNV\"]\n",
    "#display(rc.shape)\n",
    "#display(rc.head())\n",
    "\n",
    "# Optional Export to excel\n",
    "#rc.to_excel(cwd+\"/annotated_clinvar_filt_retnet_processed.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "387a3827-a4b5-4e91-ad2b-515337b0a88d",
   "metadata": {},
   "source": [
    "### MutScore Thresholds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8f75fb9b-d84e-4d07-b419-e13cb35406cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>MutScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1854</td>\n",
       "      <td>42.618384</td>\n",
       "      <td>57.381616</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  MutScore\n",
       "0  1377  1857    42.578850    57.421150     0.000\n",
       "0  1377  1857    42.578850    57.421150     0.001\n",
       "0  1377  1857    42.578850    57.421150     0.002\n",
       "0  1377  1857    42.578850    57.421150     0.003\n",
       "0  1377  1854    42.618384    57.381616     0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of MutScore is 0.868\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>MutScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  MutScore\n",
       "0    0    0          NaN          NaN     0.000\n",
       "0    0    0          NaN          NaN     0.001\n",
       "0    0    0          NaN          NaN     0.002\n",
       "0    0    3          0.0        100.0     0.003\n",
       "0    0    5          0.0        100.0     0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of MutScore is 0.558\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=0, max_score=1, df=rc, classifier=\"MutScore\")\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of MutScore is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_MutScore.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=0, max_score=1, df=rc, classifier=\"MutScore\")\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of MutScore is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_MutScore.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8966aad6-e0d0-4d4e-bd9f-d195f4efbe39",
   "metadata": {},
   "source": [
    "### VEST4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "4c0f8e8e-4254-462f-8177-cc97d9434253",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>VEST4_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.504627</td>\n",
       "      <td>57.495373</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.504627</td>\n",
       "      <td>57.495373</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.504627</td>\n",
       "      <td>57.495373</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.504627</td>\n",
       "      <td>57.495373</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.504627</td>\n",
       "      <td>57.495373</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  VEST4_score\n",
       "0  1378  1864    42.504627    57.495373        0.000\n",
       "0  1378  1864    42.504627    57.495373        0.001\n",
       "0  1378  1864    42.504627    57.495373        0.002\n",
       "0  1378  1864    42.504627    57.495373        0.003\n",
       "0  1378  1864    42.504627    57.495373        0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of VEST4_score is 0.906\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>VEST4_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  VEST4_score\n",
       "0    0    0          NaN          NaN        0.000\n",
       "0    0    0          NaN          NaN        0.001\n",
       "0    0    0          NaN          NaN        0.002\n",
       "0    0    0          NaN          NaN        0.003\n",
       "0    0    0          NaN          NaN        0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of VEST4_score is 0.463\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=0, max_score=1, df=rc, classifier=\"VEST4_score\")\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of VEST4_score is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_VEST4_score.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=0, max_score=1, df=rc, classifier=\"VEST4_score\")\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of VEST4_score is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_VEST4_score.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b50de92b-052c-4aa9-a84e-077876924531",
   "metadata": {},
   "source": [
    "### REVEL"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "bffe990f-1c0e-45ce-af43-245f44ace621",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>REVEL_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1857</td>\n",
       "      <td>42.578850</td>\n",
       "      <td>57.421150</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1856</td>\n",
       "      <td>42.592020</td>\n",
       "      <td>57.407980</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1855</td>\n",
       "      <td>42.605198</td>\n",
       "      <td>57.394802</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1377</td>\n",
       "      <td>1854</td>\n",
       "      <td>42.618384</td>\n",
       "      <td>57.381616</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  REVEL_score\n",
       "0  1377  1857    42.578850    57.421150        0.000\n",
       "0  1377  1857    42.578850    57.421150        0.001\n",
       "0  1377  1856    42.592020    57.407980        0.002\n",
       "0  1377  1855    42.605198    57.394802        0.003\n",
       "0  1377  1854    42.618384    57.381616        0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of REVEL_score is 0.796\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>REVEL_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  REVEL_score\n",
       "0    0    0          NaN          NaN        0.000\n",
       "0    0    1          0.0        100.0        0.001\n",
       "0    0    2          0.0        100.0        0.002\n",
       "0    0    3          0.0        100.0        0.003\n",
       "0    0    4          0.0        100.0        0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of REVEL_score is 0.368\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=0, max_score=1, df=rc, classifier=\"REVEL_score\")\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of REVEL_score is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_REVEL_score.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=0, max_score=1, df=rc, classifier=\"REVEL_score\")\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of REVEL_score is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_REVEL_score.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed7d84f3-4d07-4bab-ba1f-e9819cb06519",
   "metadata": {},
   "source": [
    "### ClinPred"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "27c169b1-5258-424b-a0a5-101f45dff2ad",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>ClinPred_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1859</td>\n",
       "      <td>42.570281</td>\n",
       "      <td>57.429719</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1855</td>\n",
       "      <td>42.622951</td>\n",
       "      <td>57.377049</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1838</td>\n",
       "      <td>42.848259</td>\n",
       "      <td>57.151741</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1817</td>\n",
       "      <td>43.129890</td>\n",
       "      <td>56.870110</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1378</td>\n",
       "      <td>1803</td>\n",
       "      <td>43.319711</td>\n",
       "      <td>56.680289</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  ClinPred_score\n",
       "0  1378  1859    42.570281    57.429719           0.000\n",
       "0  1378  1855    42.622951    57.377049           0.001\n",
       "0  1378  1838    42.848259    57.151741           0.002\n",
       "0  1378  1817    43.129890    56.870110           0.003\n",
       "0  1378  1803    43.319711    56.680289           0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of ClinPred_score is 0.975\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>ClinPred_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>21</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>42</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>56</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>76</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  ClinPred_score\n",
       "0    0    4          0.0        100.0           0.000\n",
       "0    0   21          0.0        100.0           0.001\n",
       "0    0   42          0.0        100.0           0.002\n",
       "0    0   56          0.0        100.0           0.003\n",
       "0    0   76          0.0        100.0           0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of ClinPred_score is 0.851\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=0, max_score=1, df=rc, classifier=\"ClinPred_score\")\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of ClinPred_score is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_ClinPred_score.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=0, max_score=1, df=rc, classifier=\"ClinPred_score\")\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of ClinPred_score is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_ClinPred_score.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3dd1b39-fa49-4a86-ac1d-d1e07be88988",
   "metadata": {},
   "source": [
    "### MetaRNN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d78ebdf4-8a14-43b2-bdb7-3e3e5eae8601",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1381</td>\n",
       "      <td>1873</td>\n",
       "      <td>42.440074</td>\n",
       "      <td>57.559926</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1381</td>\n",
       "      <td>1839</td>\n",
       "      <td>42.888199</td>\n",
       "      <td>57.111801</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1381</td>\n",
       "      <td>1824</td>\n",
       "      <td>43.088924</td>\n",
       "      <td>56.911076</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1381</td>\n",
       "      <td>1788</td>\n",
       "      <td>43.578416</td>\n",
       "      <td>56.421584</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1381</td>\n",
       "      <td>1748</td>\n",
       "      <td>44.135507</td>\n",
       "      <td>55.864493</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  MetaRNN_score\n",
       "0  1381  1873    42.440074    57.559926          0.000\n",
       "0  1381  1839    42.888199    57.111801          0.001\n",
       "0  1381  1824    43.088924    56.911076          0.002\n",
       "0  1381  1788    43.578416    56.421584          0.003\n",
       "0  1381  1748    44.135507    55.864493          0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of MetaRNN_score is 0.814\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>MetaRNN_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>34</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>49</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>85</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>125</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>193</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100.0</td>\n",
       "      <td>0.004</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  MetaRNN_score\n",
       "0    0   34          0.0        100.0          0.000\n",
       "0    0   49          0.0        100.0          0.001\n",
       "0    0   85          0.0        100.0          0.002\n",
       "0    0  125          0.0        100.0          0.003\n",
       "0    0  193          0.0        100.0          0.004"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of MetaRNN_score is 0.609\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=0, max_score=1, df=rc, classifier=\"MetaRNN_score\")\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of MetaRNN_score is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_MetaRNN_score.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=0, max_score=1, df=rc, classifier=\"MetaRNN_score\")\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of MetaRNN_score is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_MetaRNN_score.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce1db9a6-e76e-4098-a3de-566fd6692b21",
   "metadata": {},
   "source": [
    "### BayesDel_addAF_score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "ba83c120-4d07-491e-b7f5-972357e89841",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1382</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.575478</td>\n",
       "      <td>57.424522</td>\n",
       "      <td>-1.293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1382</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.575478</td>\n",
       "      <td>57.424522</td>\n",
       "      <td>-1.292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1382</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.575478</td>\n",
       "      <td>57.424522</td>\n",
       "      <td>-1.291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1382</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.575478</td>\n",
       "      <td>57.424522</td>\n",
       "      <td>-1.290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1382</td>\n",
       "      <td>1864</td>\n",
       "      <td>42.575478</td>\n",
       "      <td>57.424522</td>\n",
       "      <td>-1.289</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PLP   BLB  percent_PLP  percent_BLB  BayesDel_addAF_score\n",
       "0  1382  1864    42.575478    57.424522                -1.293\n",
       "0  1382  1864    42.575478    57.424522                -1.292\n",
       "0  1382  1864    42.575478    57.424522                -1.291\n",
       "0  1382  1864    42.575478    57.424522                -1.290\n",
       "0  1382  1864    42.575478    57.424522                -1.289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The upper likely-pathogenic threshold of BayesDel_addAF_score is 0.267\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PLP</th>\n",
       "      <th>BLB</th>\n",
       "      <th>percent_PLP</th>\n",
       "      <th>percent_BLB</th>\n",
       "      <th>BayesDel_addAF_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.292</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.291</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.289</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   PLP  BLB  percent_PLP  percent_BLB  BayesDel_addAF_score\n",
       "0    0    0          NaN          NaN                -1.293\n",
       "0    0    0          NaN          NaN                -1.292\n",
       "0    0    0          NaN          NaN                -1.291\n",
       "0    0    0          NaN          NaN                -1.290\n",
       "0    0    0          NaN          NaN                -1.289"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The lower likely-benign threshold of BayesDel_addAF_score is 0.013\n"
     ]
    }
   ],
   "source": [
    "# Likely Pathogenic thresholds\n",
    "lp_scores, lp = likely_pathogenic(min_score=-1.293, max_score=0.758, df=rc, classifier=\"BayesDel_addAF_score\", iterations=2052)\n",
    "display(lp_scores.head())\n",
    "print(\"The upper likely-pathogenic threshold of BayesDel_addAF_score is {}\".format(lp))\n",
    "#lp_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-pathogenic_BayesDel_addAF_score.xlsx\", index=False) #optional export to excel\n",
    "\n",
    "# Likely Benign thresholds\n",
    "lb_scores, lb = likely_benign(min_score=-1.293, max_score=0.758, df=rc, classifier=\"BayesDel_addAF_score\", iterations=2052)\n",
    "display(lb_scores.head())\n",
    "print(\"The lower likely-benign threshold of BayesDel_addAF_score is {}\".format(lb))\n",
    "#lb_scores.to_excel(cwd+\"/retnet_classifier_thresholds/likely-benign_BayesDel_addAF_score.xlsx\", index=False) #optional export to excel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e1ce4cf-e175-43e3-81dd-1e7cff5dff9b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fb0ccc0-1a1a-451b-9e7a-5ae1014fd8d4",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "82b23511-b986-4284-97e1-15e4e12832e7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7cae348b-1718-4546-94a8-68e56f62886b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "119a83ed-3002-480e-b5cc-d9298368f9aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "94872980-4f48-41f3-90ce-5a3c93d0c715",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71bc2d1c-46d1-472d-affa-1d5e0bda904e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
