{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "7c6a1e52-2a2b-478f-935c-b668407c89ba",
   "metadata": {
    "tags": []
   },
   "source": [
    "# **ClinVar SNV ROC & AUC Generator**\n",
    "#### Author: Daniel Brock\n",
    "#### Date: 2023-07-06\n",
    "#### Purpose: to re-add the \"pathogenicity rankings (PLP, BLB)\" and other columns provided in ClinVar's new variant annotated output from annovar and calculate ROC and AUC"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0b1af4bf-b953-47e6-8d18-fe8a4868cb3d",
   "metadata": {},
   "source": [
    "## **Importing Required Packages**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "32149989-434c-4bbf-9720-cd983a00e2f3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c:\\Users\\TooFastDan\\OneDrive - Baylor College of Medicine\\BCM\\Projects\\Autosomal Dominant Predictor of IRDs\\manuscript\\GitHub\n"
     ]
    }
   ],
   "source": [
    "# Importing required packages\n",
    "import os\n",
    "import sys\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt\n",
    "from sklearn.metrics import roc_curve, auc\n",
    "\n",
    "# Working directory\n",
    "cwd = os.getcwd()\n",
    "print(cwd)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1c4ba1e2-9805-4235-b2de-81b5f672ea86",
   "metadata": {},
   "source": [
    "## **Defining Important Functions for ROC-AUC Analysis**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "fedbecfa-612b-4c2f-97a9-4872d3489f73",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def roc_auc(df, model, truth, disp_results):\n",
    "    \"\"\"\n",
    "    Function to:\n",
    "    - filter NaN values and calculate percent retained\n",
    "    - calculate ROC parametrics: false positive rate (fpr), true positive rate (tpr), thresholds\n",
    "    - Area Under the Curve (auc_score)\n",
    "    \n",
    "    Inputs:\n",
    "    df: autosomal recessive or dominant dataframes\n",
    "    model: column name of the predictor model in the df\n",
    "    truth: column name for the ground truth (PLP vs BLB)\n",
    "    disp_results = True/False on whether to show results of model or not\n",
    "    \n",
    "    Outputs:\n",
    "    fpr: false positive rate\n",
    "    tpr: true positive rate\n",
    "    thresholds\n",
    "    auc_score\n",
    "    percent of SNVs retained after dropping NaN values\n",
    "    \"\"\"\n",
    "    # Filtering out NaNs and rows with characters in a specified column (model) and calculating percent retained\n",
    "    df_filt = df[~df[model].isna()]\n",
    "    percent_retained = (df_filt.shape[0] / df.shape[0]) * 100\n",
    "    \n",
    "    try:\n",
    "        # Calulating false positive rate (fpr), true positive rate (tpr), thresholds, and AUC scores from fpr + tpr\n",
    "        model_scores = df_filt[model].astype(float)\n",
    "        fpr, tpr, threshold = roc_curve(y_true=df_filt[truth], y_score=model_scores)\n",
    "        \n",
    "        # Calulating AUC scores\n",
    "        auc_score = auc(fpr, tpr)\n",
    "        \n",
    "        # Print and returning results\n",
    "        if disp_results is True:\n",
    "            print(\"{0} AUC = {1} while using {2}% of the values.\".format(model, np.round(auc_score, 5), np.round(percent_retained, 3)))\n",
    "        if disp_results is False:\n",
    "            pass\n",
    "        return fpr, tpr, threshold, auc_score, percent_retained\n",
    "    except:\n",
    "        print(\"{} failed due to mixed datatypes\".format(model))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "5e9cb161-aa39-4559-bee9-22cae5d6a08a",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "def roc_auc_looper(df_input, disp_results):\n",
    "    \"\"\"\n",
    "    Function to loop over 39 predictive models and return ROC and AUC dataframes consisting of fpr, tpr, thresholds, and AUCs.\n",
    "    Dependent on the \"roc_auc\" function.\n",
    "    \n",
    "    inputs:\n",
    "    - df_input = dataframe from annovar output with filtered genes of interest, model results, and a \"y_test\" truth column\n",
    "    - disp_results = True/False on whether to show results of model or not\n",
    "    \n",
    "    outputs: \n",
    "    - auc_df\n",
    "    - roc_df\n",
    "    \"\"\"\n",
    "    # List of models to calculate ROC & AUC values\n",
    "    models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "              \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "              \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "              \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "              \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "              \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "              \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "              \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "    \n",
    "    try:\n",
    "        # Looping and creating lists/dfs to append results to\n",
    "        auc_data = []\n",
    "        roc_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "        for m in models:\n",
    "            # Calulating parametrics for each prediction model\n",
    "            fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=df_input, model=m, truth=\"y_test\", disp_results=disp_results)\n",
    "\n",
    "            # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "            auc_data.append((m, auc_score, percent_retained))\n",
    "\n",
    "            # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "            df_temp = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "            roc_df = pd.concat([roc_df, df_temp], axis=0)\n",
    "\n",
    "        # Making the AUC dataframe from list of tuples\n",
    "        auc_df = pd.DataFrame(auc_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"])\n",
    "        auc_df = auc_df.sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "        # Displaying and returning results results\n",
    "        print(\"\\nAUC df shape: {}\".format(auc_df.shape))\n",
    "        display(auc_df.head())\n",
    "        print(\"\\nROC df shape: {}\".format(roc_df.shape))\n",
    "        display(roc_df.head())\n",
    "        return auc_df, roc_df\n",
    "    except:\n",
    "        print(\"Error: Bad input\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ea598f93-16b0-4b78-a10f-fda6739c7174",
   "metadata": {},
   "source": [
    "## **Importing the filtered clinvar dataset (data after 11/21/2020)**\n",
    "### (This was the input to ANNOVAR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "6d8a9947-d0a8-478f-a782-53ed1563b251",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62696, 24)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>CLNVC</th>\n",
       "      <th>CLNVCSO</th>\n",
       "      <th>GENEINFO</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>MC</th>\n",
       "      <th>ORIGIN</th>\n",
       "      <th>CLNSIG</th>\n",
       "      <th>PATHOGENICITY</th>\n",
       "      <th>y_test</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...</td>\n",
       "      <td>2506469</td>\n",
       "      <td>2671427</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>HK1:3098</td>\n",
       "      <td>HK1</td>\n",
       "      <td>3098</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...</td>\n",
       "      <td>2506465</td>\n",
       "      <td>2671423</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>JAG1:182</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>182</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>65300662</td>\n",
       "      <td>65300662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...</td>\n",
       "      <td>2506396</td>\n",
       "      <td>2671350</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>EYS:346007</td>\n",
       "      <td>EYS</td>\n",
       "      <td>346007</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>39255536</td>\n",
       "      <td>39255536</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...</td>\n",
       "      <td>2506391</td>\n",
       "      <td>2671345</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>WDR19:57728</td>\n",
       "      <td>WDR19</td>\n",
       "      <td>57728</td>\n",
       "      <td>SO:0001587|nonsense</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...</td>\n",
       "      <td>2506270</td>\n",
       "      <td>2671226</td>\n",
       "      <td>...</td>\n",
       "      <td>single_nucleotide_variant</td>\n",
       "      <td>SO:0001483</td>\n",
       "      <td>COL2A1:1280</td>\n",
       "      <td>COL2A1</td>\n",
       "      <td>1280</td>\n",
       "      <td>SO:0001583|missense_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely_pathogenic</td>\n",
       "      <td>PLP</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Ref Alt QUAL FILTER  \\\n",
       "0  chr10  71142347  71142347   C   A    .      .   \n",
       "1  chr20  10654126  10654126   A   C    .      .   \n",
       "2   chr6  65300662  65300662   T   A    .      .   \n",
       "3   chr4  39255536  39255536   C   T    .      .   \n",
       "4  chr12  48380959  48380959   C   T    .      .   \n",
       "\n",
       "                                                INFO       ID  ALLELEID  ...  \\\n",
       "0  ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...  2506469   2671427  ...   \n",
       "1  ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...  2506465   2671423  ...   \n",
       "2  ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...  2506396   2671350  ...   \n",
       "3  ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...  2506391   2671345  ...   \n",
       "4  ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...  2506270   2671226  ...   \n",
       "\n",
       "                       CLNVC     CLNVCSO     GENEINFO    Gene Gene_ID  \\\n",
       "0  single_nucleotide_variant  SO:0001483     HK1:3098     HK1    3098   \n",
       "1  single_nucleotide_variant  SO:0001483     JAG1:182    JAG1     182   \n",
       "2  single_nucleotide_variant  SO:0001483   EYS:346007     EYS  346007   \n",
       "3  single_nucleotide_variant  SO:0001483  WDR19:57728   WDR19   57728   \n",
       "4  single_nucleotide_variant  SO:0001483  COL2A1:1280  COL2A1    1280   \n",
       "\n",
       "                            MC ORIGIN             CLNSIG PATHOGENICITY y_test  \n",
       "0  SO:0001583|missense_variant    NaN  Likely_pathogenic           PLP      1  \n",
       "1  SO:0001583|missense_variant    NaN  Likely_pathogenic           PLP      1  \n",
       "2          SO:0001587|nonsense    NaN         Pathogenic           PLP      1  \n",
       "3          SO:0001587|nonsense    NaN  Likely_pathogenic           PLP      1  \n",
       "4  SO:0001583|missense_variant    NaN  Likely_pathogenic           PLP      1  \n",
       "\n",
       "[5 rows x 24 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Importing the new testing dataset with pathogenicity ranks and dates filtered after MutScore paper (11-21-2020 - from ClinVar_VCF_parser)\n",
    "clinvar_avi = pd.read_table(cwd+\"/annovar_files/clinvar_filt_retnet.avinput\", header=None,\n",
    "                           names=['Chr', 'Start', 'End', 'Ref', 'Alt', 'QUAL', 'FILTER', 'INFO', 'ID', 'ALLELEID', 'CLNDISDB', 'CLNDN', 'CLNHGVS', \n",
    "                                  'CLNREVSTAT', 'CLNVC', 'CLNVCSO', 'GENEINFO', 'Gene', 'Gene_ID', 'MC', 'ORIGIN', 'CLNSIG', 'PATHOGENICITY', 'y_test'])\n",
    "\n",
    "# Displaying info\n",
    "display(clinvar_avi.shape)\n",
    "display(clinvar_avi.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e89cdd4e-cff1-479d-aa6f-6f6936faa80c",
   "metadata": {},
   "source": [
    "## **Importing annotated clinvar retnet file after being run in ANNOVAR**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3a4e98e6-19fc-4ed2-b0a9-9797c502f0ed",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\TooFastDan\\AppData\\Local\\Temp\\ipykernel_4256\\3358280549.py:2: DtypeWarning: Columns (118,119) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  anno = pd.read_csv(cwd+\"/annovar_files/clinvar_filt_retnet_annotated.hg19_multianno.csv\", na_values=\".\")\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(62696, 121)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>Func.refGene</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>GeneDetail.refGene</th>\n",
       "      <th>ExonicFunc.refGene</th>\n",
       "      <th>AAChange.refGene</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>GTEx_V8_gene</th>\n",
       "      <th>GTEx_V8_tissue</th>\n",
       "      <th>MutScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>HK1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>HK1:NM_001322367:exon9:c.C1274A:p.T425K,HK1:NM...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.826</td>\n",
       "      <td>0.344</td>\n",
       "      <td>20.152</td>\n",
       "      <td>0.981</td>\n",
       "      <td>Hexokinase\\x2c C-terminal;Hexokinase\\x2c C-ter...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>exonic</td>\n",
       "      <td>JAG1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>JAG1:NM_000214:exon1:c.T53G:p.L18R</td>\n",
       "      <td>...</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.413</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.863</td>\n",
       "      <td>12.448</td>\n",
       "      <td>0.550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>65300662</td>\n",
       "      <td>65300662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>exonic</td>\n",
       "      <td>EYS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>EYS:NM_001142800:exon26:c.A5098T:p.K1700X,EYS:...</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014</td>\n",
       "      <td>0.190</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.659</td>\n",
       "      <td>5.923</td>\n",
       "      <td>0.183</td>\n",
       "      <td>.;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>39255536</td>\n",
       "      <td>39255536</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>WDR19</td>\n",
       "      <td>NaN</td>\n",
       "      <td>stopgain</td>\n",
       "      <td>WDR19:NM_001317924:exon25:c.C2407T:p.Q803X,WDR...</td>\n",
       "      <td>...</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.402</td>\n",
       "      <td>0.986</td>\n",
       "      <td>0.501</td>\n",
       "      <td>14.372</td>\n",
       "      <td>0.664</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>exonic</td>\n",
       "      <td>COL2A1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>nonsynonymous SNV</td>\n",
       "      <td>COL2A1:NM_033150:exon20:c.G1060A:p.G354S,COL2A...</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.967</td>\n",
       "      <td>0.444</td>\n",
       "      <td>16.928</td>\n",
       "      <td>0.861</td>\n",
       "      <td>.;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.993</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 121 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Ref Alt Func.refGene Gene.refGene  \\\n",
       "0  chr10  71142347  71142347   C   A       exonic          HK1   \n",
       "1  chr20  10654126  10654126   A   C       exonic         JAG1   \n",
       "2   chr6  65300662  65300662   T   A       exonic          EYS   \n",
       "3   chr4  39255536  39255536   C   T       exonic        WDR19   \n",
       "4  chr12  48380959  48380959   C   T       exonic       COL2A1   \n",
       "\n",
       "  GeneDetail.refGene ExonicFunc.refGene  \\\n",
       "0                NaN  nonsynonymous SNV   \n",
       "1                NaN  nonsynonymous SNV   \n",
       "2                NaN           stopgain   \n",
       "3                NaN           stopgain   \n",
       "4                NaN  nonsynonymous SNV   \n",
       "\n",
       "                                    AAChange.refGene  ...  \\\n",
       "0  HK1:NM_001322367:exon9:c.C1274A:p.T425K,HK1:NM...  ...   \n",
       "1                 JAG1:NM_000214:exon1:c.T53G:p.L18R  ...   \n",
       "2  EYS:NM_001142800:exon26:c.A5098T:p.K1700X,EYS:...  ...   \n",
       "3  WDR19:NM_001317924:exon25:c.C2407T:p.Q803X,WDR...  ...   \n",
       "4  COL2A1:NM_033150:exon20:c.G1060A:p.G354S,COL2A...  ...   \n",
       "\n",
       "  phastCons100way_vertebrate  phastCons100way_vertebrate_rankscore  \\\n",
       "0                      1.000                                 0.716   \n",
       "1                      0.998                                 0.413   \n",
       "2                      0.014                                 0.190   \n",
       "3                      0.997                                 0.402   \n",
       "4                      1.000                                 0.716   \n",
       "\n",
       "   phastCons30way_mammalian  phastCons30way_mammalian_rankscore  \\\n",
       "0                     0.826                               0.344   \n",
       "1                     1.000                               0.863   \n",
       "2                     0.998                               0.659   \n",
       "3                     0.986                               0.501   \n",
       "4                     0.967                               0.444   \n",
       "\n",
       "   SiPhy_29way_logOdds SiPhy_29way_logOdds_rankscore  \\\n",
       "0               20.152                         0.981   \n",
       "1               12.448                         0.550   \n",
       "2                5.923                         0.183   \n",
       "3               14.372                         0.664   \n",
       "4               16.928                         0.861   \n",
       "\n",
       "                                     Interpro_domain  GTEx_V8_gene  \\\n",
       "0  Hexokinase\\x2c C-terminal;Hexokinase\\x2c C-ter...           NaN   \n",
       "1                                                NaN           NaN   \n",
       "2                                                .;.           NaN   \n",
       "3                                                NaN           NaN   \n",
       "4                                                .;.           NaN   \n",
       "\n",
       "  GTEx_V8_tissue  MutScore  \n",
       "0            NaN     0.734  \n",
       "1            NaN     0.700  \n",
       "2            NaN       NaN  \n",
       "3            NaN       NaN  \n",
       "4            NaN     0.993  \n",
       "\n",
       "[5 rows x 121 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Importing the annotated csv file from annovar (run in linux)\n",
    "anno = pd.read_csv(cwd+\"/annovar_files/clinvar_filt_retnet_annotated.hg19_multianno.csv\", na_values=\".\")\n",
    "display(anno.shape)\n",
    "display(anno.head())"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1f217c85-f89c-4b00-8f66-0c6af6c92b5f",
   "metadata": {},
   "source": [
    "## **Merging the two datasets to include pathogenicity scoring**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "2589c542-fd21-48fb-a6c3-34d13c4a26d1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(62693, 140)"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>phastCons100way_vertebrate</th>\n",
       "      <th>phastCons100way_vertebrate_rankscore</th>\n",
       "      <th>phastCons30way_mammalian</th>\n",
       "      <th>phastCons30way_mammalian_rankscore</th>\n",
       "      <th>SiPhy_29way_logOdds</th>\n",
       "      <th>SiPhy_29way_logOdds_rankscore</th>\n",
       "      <th>Interpro_domain</th>\n",
       "      <th>GTEx_V8_gene</th>\n",
       "      <th>GTEx_V8_tissue</th>\n",
       "      <th>MutScore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...</td>\n",
       "      <td>2506469</td>\n",
       "      <td>2671427</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.826</td>\n",
       "      <td>0.344</td>\n",
       "      <td>20.152</td>\n",
       "      <td>0.981</td>\n",
       "      <td>Hexokinase\\x2c C-terminal;Hexokinase\\x2c C-ter...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.734</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...</td>\n",
       "      <td>2506465</td>\n",
       "      <td>2671423</td>\n",
       "      <td>...</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.413</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.863</td>\n",
       "      <td>12.448</td>\n",
       "      <td>0.550</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.700</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6</td>\n",
       "      <td>65300662</td>\n",
       "      <td>65300662</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...</td>\n",
       "      <td>2506396</td>\n",
       "      <td>2671350</td>\n",
       "      <td>...</td>\n",
       "      <td>0.014</td>\n",
       "      <td>0.190</td>\n",
       "      <td>0.998</td>\n",
       "      <td>0.659</td>\n",
       "      <td>5.923</td>\n",
       "      <td>0.183</td>\n",
       "      <td>.;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr4</td>\n",
       "      <td>39255536</td>\n",
       "      <td>39255536</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...</td>\n",
       "      <td>2506391</td>\n",
       "      <td>2671345</td>\n",
       "      <td>...</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.402</td>\n",
       "      <td>0.986</td>\n",
       "      <td>0.501</td>\n",
       "      <td>14.372</td>\n",
       "      <td>0.664</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...</td>\n",
       "      <td>2506270</td>\n",
       "      <td>2671226</td>\n",
       "      <td>...</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.716</td>\n",
       "      <td>0.967</td>\n",
       "      <td>0.444</td>\n",
       "      <td>16.928</td>\n",
       "      <td>0.861</td>\n",
       "      <td>.;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.993</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 140 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Ref Alt QUAL FILTER  \\\n",
       "0  chr10  71142347  71142347   C   A    .      .   \n",
       "1  chr20  10654126  10654126   A   C    .      .   \n",
       "2   chr6  65300662  65300662   T   A    .      .   \n",
       "3   chr4  39255536  39255536   C   T    .      .   \n",
       "4  chr12  48380959  48380959   C   T    .      .   \n",
       "\n",
       "                                                INFO       ID  ALLELEID  ...  \\\n",
       "0  ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...  2506469   2671427  ...   \n",
       "1  ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...  2506465   2671423  ...   \n",
       "2  ALLELEID=2671350;CLNDISDB=Human_Phenotype_Onto...  2506396   2671350  ...   \n",
       "3  ALLELEID=2671345;CLNDISDB=MONDO:MONDO:0013719,...  2506391   2671345  ...   \n",
       "4  ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...  2506270   2671226  ...   \n",
       "\n",
       "  phastCons100way_vertebrate phastCons100way_vertebrate_rankscore  \\\n",
       "0                      1.000                                0.716   \n",
       "1                      0.998                                0.413   \n",
       "2                      0.014                                0.190   \n",
       "3                      0.997                                0.402   \n",
       "4                      1.000                                0.716   \n",
       "\n",
       "  phastCons30way_mammalian phastCons30way_mammalian_rankscore  \\\n",
       "0                    0.826                              0.344   \n",
       "1                    1.000                              0.863   \n",
       "2                    0.998                              0.659   \n",
       "3                    0.986                              0.501   \n",
       "4                    0.967                              0.444   \n",
       "\n",
       "  SiPhy_29way_logOdds SiPhy_29way_logOdds_rankscore  \\\n",
       "0              20.152                         0.981   \n",
       "1              12.448                         0.550   \n",
       "2               5.923                         0.183   \n",
       "3              14.372                         0.664   \n",
       "4              16.928                         0.861   \n",
       "\n",
       "                                     Interpro_domain GTEx_V8_gene  \\\n",
       "0  Hexokinase\\x2c C-terminal;Hexokinase\\x2c C-ter...          NaN   \n",
       "1                                                NaN          NaN   \n",
       "2                                                .;.          NaN   \n",
       "3                                                NaN          NaN   \n",
       "4                                                .;.          NaN   \n",
       "\n",
       "  GTEx_V8_tissue MutScore  \n",
       "0            NaN    0.734  \n",
       "1            NaN    0.700  \n",
       "2            NaN      NaN  \n",
       "3            NaN      NaN  \n",
       "4            NaN    0.993  \n",
       "\n",
       "[5 rows x 140 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "anno_clinvar = pd.merge(clinvar_avi, anno, on=[\"Chr\", \"Start\", \"End\", \"Ref\", \"Alt\"]) #merging\n",
    "anno_clinvar = anno_clinvar.drop_duplicates() #dropping a small number of duplicates\n",
    "display(anno_clinvar.shape)\n",
    "display(anno_clinvar.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "23ca1d9a-dd17-4f3d-b5b2-fcc307d470b1",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Optional export of the merged df\n",
    "#anno_clinvar.to_csv(cwd+\"/annovar_files/clinvar_filt_retnet_annotated_PLP-BLB.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5fd8718f-b96a-45e0-8fc8-20218e47cc3d",
   "metadata": {},
   "source": [
    "## **Filtering for only nonsynonomous SNVs**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "31954d8c-a8e1-4f21-aa77-3d28a88a49f3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shape of ClinVar variants filtered for nonsynonomous variants: (3322, 140)\n",
      "Shape of total ClinVar IRD variants: (62693, 140)\n",
      "5.3% nonsynonomous SNV representation\n"
     ]
    }
   ],
   "source": [
    "anno_clinvar_filt = anno_clinvar[anno_clinvar[\"ExonicFunc.refGene\"]==\"nonsynonymous SNV\"]\n",
    "ird_total_variants = anno_clinvar.shape[0]\n",
    "ird_nonsyn_variants = anno_clinvar_filt.shape[0]\n",
    "percent_nonsyn_IRD = np.round((ird_nonsyn_variants / ird_total_variants) * 100, decimals=2)\n",
    "print(\"Shape of ClinVar variants filtered for nonsynonomous variants: {0}\\nShape of total ClinVar IRD variants: {1}\\n{2}% nonsynonomous SNV representation\".format(\n",
    "    str(anno_clinvar_filt.shape), str(anno_clinvar.shape), str(percent_nonsyn_IRD)))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f21aa3b9-b706-42d2-ba04-edb2cc766f02",
   "metadata": {},
   "source": [
    "## **Separating filtered, nonsynonomous IRD genes based on inheritance pattern defined in RetNet**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "371c24b1-9de2-4764-966c-afc806d1ae0e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Symbols;\\nOMIM Numbers</th>\n",
       "      <th>Gene.refGene</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>Location</th>\n",
       "      <th>Diseases;_x000D_\\nProtein</th>\n",
       "      <th>OMIM Phenotypes</th>\n",
       "      <th>How Identified;_x000D_\\nComments</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>pLI</th>\n",
       "      <th>exac_pLI</th>\n",
       "      <th>AR</th>\n",
       "      <th>AD</th>\n",
       "      <th>AD GOF</th>\n",
       "      <th>AD Haploinsuffiency</th>\n",
       "      <th>Complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL11A1, STL2;_x000D_\\n 120280, 154780, 604841</td>\n",
       "      <td>COL11A1</td>\n",
       "      <td>ENSG00000060718</td>\n",
       "      <td>1p21.1</td>\n",
       "      <td>dominant Stickler syndrome, type II; dominant ...</td>\n",
       "      <td>Fibrochondrogenesis 1, 228520 (3), Autosomal r...</td>\n",
       "      <td>linkage mapping, candidate gene; Stickler synd...</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.99998</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PRPF3, HPRP3, PRP3, RP18;_x000D_\\n 268000, 601...</td>\n",
       "      <td>PRPF3</td>\n",
       "      <td>ENSG00000117360</td>\n",
       "      <td>1q21.2</td>\n",
       "      <td>dominant retinitis pigmentosa; protein: pre-mR...</td>\n",
       "      <td>Retinitis pigmentosa 18, 601414 (3), Autosomal...</td>\n",
       "      <td>linkage mapping, candidate gene; English and D...</td>\n",
       "      <td>1</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MFN2, CMT6, CMT2A2, MARF;\\n 608507, 609260, 60...</td>\n",
       "      <td>MFN2</td>\n",
       "      <td>ENSG00000116688</td>\n",
       "      <td>1p36.22</td>\n",
       "      <td>dominant optic atrophy with neuropathy and myo...</td>\n",
       "      <td>Lipomatosis, multiple symmetric, with or witho...</td>\n",
       "      <td>candidate gene; dominant mutation in a large T...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.99415</td>\n",
       "      <td>0.99982</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ELOVL1;_x000D_\\n 611813</td>\n",
       "      <td>ELOVL1</td>\n",
       "      <td>ENSG00000066322</td>\n",
       "      <td>1p34.2</td>\n",
       "      <td>dominant optic atrophy, deafness, ichthyosis a...</td>\n",
       "      <td>Ichthyotic keratoderma, spasticity, hypomyelin...</td>\n",
       "      <td>whole-exome sequencing; identical, de novo, do...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.67897</td>\n",
       "      <td>0.22341</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ADIPOR1, PAQR1;_x000D_\\n 607945</td>\n",
       "      <td>ADIPOR1</td>\n",
       "      <td>ENSG00000159346</td>\n",
       "      <td>1q32.1</td>\n",
       "      <td>recessive retinitis pigmentosa, syndromic, Bar...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>whole-exome sequencing; a homozygous frameshif...</td>\n",
       "      <td>1</td>\n",
       "      <td>0.65840</td>\n",
       "      <td>0.59275</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Symbols;\\nOMIM Numbers Gene.refGene  \\\n",
       "0     COL11A1, STL2;_x000D_\\n 120280, 154780, 604841      COL11A1   \n",
       "1  PRPF3, HPRP3, PRP3, RP18;_x000D_\\n 268000, 601...        PRPF3   \n",
       "2  MFN2, CMT6, CMT2A2, MARF;\\n 608507, 609260, 60...         MFN2   \n",
       "3                            ELOVL1;_x000D_\\n 611813       ELOVL1   \n",
       "4                    ADIPOR1, PAQR1;_x000D_\\n 607945      ADIPOR1   \n",
       "\n",
       "           gene_id Location  \\\n",
       "0  ENSG00000060718   1p21.1   \n",
       "1  ENSG00000117360   1q21.2   \n",
       "2  ENSG00000116688  1p36.22   \n",
       "3  ENSG00000066322   1p34.2   \n",
       "4  ENSG00000159346   1q32.1   \n",
       "\n",
       "                           Diseases;_x000D_\\nProtein  \\\n",
       "0  dominant Stickler syndrome, type II; dominant ...   \n",
       "1  dominant retinitis pigmentosa; protein: pre-mR...   \n",
       "2  dominant optic atrophy with neuropathy and myo...   \n",
       "3  dominant optic atrophy, deafness, ichthyosis a...   \n",
       "4  recessive retinitis pigmentosa, syndromic, Bar...   \n",
       "\n",
       "                                     OMIM Phenotypes  \\\n",
       "0  Fibrochondrogenesis 1, 228520 (3), Autosomal r...   \n",
       "1  Retinitis pigmentosa 18, 601414 (3), Autosomal...   \n",
       "2  Lipomatosis, multiple symmetric, with or witho...   \n",
       "3  Ichthyotic keratoderma, spasticity, hypomyelin...   \n",
       "4                                                NaN   \n",
       "\n",
       "                    How Identified;_x000D_\\nComments Chromosome      pLI  \\\n",
       "0  linkage mapping, candidate gene; Stickler synd...          1  1.00000   \n",
       "1  linkage mapping, candidate gene; English and D...          1  1.00000   \n",
       "2  candidate gene; dominant mutation in a large T...          1  0.99415   \n",
       "3  whole-exome sequencing; identical, de novo, do...          1  0.67897   \n",
       "4  whole-exome sequencing; a homozygous frameshif...          1  0.65840   \n",
       "\n",
       "   exac_pLI  AR  AD  AD GOF  AD Haploinsuffiency  Complex  \n",
       "0   0.99998   0   1       0                    1      NaN  \n",
       "1   1.00000   0   1       0                    1      NaN  \n",
       "2   0.99982   0   1       0                    1      NaN  \n",
       "3   0.22341   0   1       1                    0      NaN  \n",
       "4   0.59275   1   1       1                    0      1.0  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELEID</th>\n",
       "      <th>...</th>\n",
       "      <th>OMIM Phenotypes</th>\n",
       "      <th>How Identified;_x000D_\\nComments</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>pLI</th>\n",
       "      <th>exac_pLI</th>\n",
       "      <th>AR</th>\n",
       "      <th>AD</th>\n",
       "      <th>AD GOF</th>\n",
       "      <th>AD Haploinsuffiency</th>\n",
       "      <th>Complex</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr10</td>\n",
       "      <td>71142347</td>\n",
       "      <td>71142347</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...</td>\n",
       "      <td>2506469</td>\n",
       "      <td>2671427</td>\n",
       "      <td>...</td>\n",
       "      <td>Retinitis pigmentosa 79, 617460 (3), Autosomal...</td>\n",
       "      <td>linkage mapping, whole-exome sequencing; a mis...</td>\n",
       "      <td>10</td>\n",
       "      <td>9.145900e-01</td>\n",
       "      <td>5.557300e-01</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr20</td>\n",
       "      <td>10654126</td>\n",
       "      <td>10654126</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...</td>\n",
       "      <td>2506465</td>\n",
       "      <td>2671423</td>\n",
       "      <td>...</td>\n",
       "      <td>?Deafness, congenital heart defects, and poste...</td>\n",
       "      <td>deletion mapping, candidate gene; multiple aff...</td>\n",
       "      <td>20</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48380959</td>\n",
       "      <td>48380959</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...</td>\n",
       "      <td>2506270</td>\n",
       "      <td>2671226</td>\n",
       "      <td>...</td>\n",
       "      <td>?Vitreoretinopathy with phalangeal epiphyseal ...</td>\n",
       "      <td>linkage mapping, candidate gene; mutations in ...</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr20</td>\n",
       "      <td>3897663</td>\n",
       "      <td>3897663</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2671068;CLNDISDB=MONDO:MONDO:0009319,...</td>\n",
       "      <td>2506108</td>\n",
       "      <td>2671068</td>\n",
       "      <td>...</td>\n",
       "      <td>HARP syndrome, 607236 (3), Autosomal recessive...</td>\n",
       "      <td>homozygosity mapping, candidate gene; symptoms...</td>\n",
       "      <td>20</td>\n",
       "      <td>7.403400e-07</td>\n",
       "      <td>1.079200e-02</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr12</td>\n",
       "      <td>48378867</td>\n",
       "      <td>48378867</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=2670853;CLNDISDB=MedGen:CN517202;CLND...</td>\n",
       "      <td>2505889</td>\n",
       "      <td>2670853</td>\n",
       "      <td>...</td>\n",
       "      <td>?Vitreoretinopathy with phalangeal epiphyseal ...</td>\n",
       "      <td>linkage mapping, candidate gene; mutations in ...</td>\n",
       "      <td>12</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3325</th>\n",
       "      <td>chr11</td>\n",
       "      <td>86662324</td>\n",
       "      <td>86662324</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=975335;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "      <td>986930</td>\n",
       "      <td>975335</td>\n",
       "      <td>...</td>\n",
       "      <td>Retinopathy of prematurity, 133780 (3), Autoso...</td>\n",
       "      <td>linkage mapping, candidate gene; Criswick-Sche...</td>\n",
       "      <td>11</td>\n",
       "      <td>9.696200e-01</td>\n",
       "      <td>9.207900e-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3326</th>\n",
       "      <td>chr1</td>\n",
       "      <td>94506874</td>\n",
       "      <td>94506874</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=974993;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "      <td>986929</td>\n",
       "      <td>974993</td>\n",
       "      <td>...</td>\n",
       "      <td>Retinal dystrophy, early-onset severe, 248200 ...</td>\n",
       "      <td>linkage mapping, candidate gene; may be involv...</td>\n",
       "      <td>1</td>\n",
       "      <td>5.349900e-48</td>\n",
       "      <td>1.267600e-28</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3327</th>\n",
       "      <td>chr1</td>\n",
       "      <td>103354176</td>\n",
       "      <td>103354176</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=974943;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "      <td>986890</td>\n",
       "      <td>974943</td>\n",
       "      <td>...</td>\n",
       "      <td>Fibrochondrogenesis 1, 228520 (3), Autosomal r...</td>\n",
       "      <td>linkage mapping, candidate gene; Stickler synd...</td>\n",
       "      <td>1</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>9.999800e-01</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3328</th>\n",
       "      <td>chr6</td>\n",
       "      <td>42146081</td>\n",
       "      <td>42146081</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=975196;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "      <td>986883</td>\n",
       "      <td>975196</td>\n",
       "      <td>...</td>\n",
       "      <td>Cone-rod dystrophy 14, 602093 (3), Autosomal d...</td>\n",
       "      <td>linkage mapping, candidate gene; British famil...</td>\n",
       "      <td>6</td>\n",
       "      <td>4.142900e-04</td>\n",
       "      <td>9.918300e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3329</th>\n",
       "      <td>chr9</td>\n",
       "      <td>2718950</td>\n",
       "      <td>2718950</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>ALLELEID=975268;CLNDISDB=MedGen:CN517202;CLNDN...</td>\n",
       "      <td>986878</td>\n",
       "      <td>975268</td>\n",
       "      <td>...</td>\n",
       "      <td>Retinal cone dystrophy 3B, 610356 (3), Autosom...</td>\n",
       "      <td>homozygosity mapping, candidate gene; mutation...</td>\n",
       "      <td>9</td>\n",
       "      <td>1.194400e-26</td>\n",
       "      <td>6.271100e-17</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3330 rows × 154 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Chr      Start        End Ref Alt QUAL FILTER  \\\n",
       "0     chr10   71142347   71142347   C   A    .      .   \n",
       "1     chr20   10654126   10654126   A   C    .      .   \n",
       "2     chr12   48380959   48380959   C   T    .      .   \n",
       "3     chr20    3897663    3897663   T   A    .      .   \n",
       "4     chr12   48378867   48378867   C   T    .      .   \n",
       "...     ...        ...        ...  ..  ..  ...    ...   \n",
       "3325  chr11   86662324   86662324   C   G    .      .   \n",
       "3326   chr1   94506874   94506874   A   T    .      .   \n",
       "3327   chr1  103354176  103354176   C   A    .      .   \n",
       "3328   chr6   42146081   42146081   G   A    .      .   \n",
       "3329   chr9    2718950    2718950   T   C    .      .   \n",
       "\n",
       "                                                   INFO       ID  ALLELEID  \\\n",
       "0     ALLELEID=2671427;CLNDISDB=MONDO:MONDO:0032807,...  2506469   2671427   \n",
       "1     ALLELEID=2671423;CLNDISDB=MONDO:MONDO:0016862,...  2506465   2671423   \n",
       "2     ALLELEID=2671226;CLNDISDB=MONDO:MONDO:0008702,...  2506270   2671226   \n",
       "3     ALLELEID=2671068;CLNDISDB=MONDO:MONDO:0009319,...  2506108   2671068   \n",
       "4     ALLELEID=2670853;CLNDISDB=MedGen:CN517202;CLND...  2505889   2670853   \n",
       "...                                                 ...      ...       ...   \n",
       "3325  ALLELEID=975335;CLNDISDB=MedGen:CN517202;CLNDN...   986930    975335   \n",
       "3326  ALLELEID=974993;CLNDISDB=MedGen:CN517202;CLNDN...   986929    974993   \n",
       "3327  ALLELEID=974943;CLNDISDB=MedGen:CN517202;CLNDN...   986890    974943   \n",
       "3328  ALLELEID=975196;CLNDISDB=MedGen:CN517202;CLNDN...   986883    975196   \n",
       "3329  ALLELEID=975268;CLNDISDB=MedGen:CN517202;CLNDN...   986878    975268   \n",
       "\n",
       "      ...                                    OMIM Phenotypes  \\\n",
       "0     ...  Retinitis pigmentosa 79, 617460 (3), Autosomal...   \n",
       "1     ...  ?Deafness, congenital heart defects, and poste...   \n",
       "2     ...  ?Vitreoretinopathy with phalangeal epiphyseal ...   \n",
       "3     ...  HARP syndrome, 607236 (3), Autosomal recessive...   \n",
       "4     ...  ?Vitreoretinopathy with phalangeal epiphyseal ...   \n",
       "...   ...                                                ...   \n",
       "3325  ...  Retinopathy of prematurity, 133780 (3), Autoso...   \n",
       "3326  ...  Retinal dystrophy, early-onset severe, 248200 ...   \n",
       "3327  ...  Fibrochondrogenesis 1, 228520 (3), Autosomal r...   \n",
       "3328  ...  Cone-rod dystrophy 14, 602093 (3), Autosomal d...   \n",
       "3329  ...  Retinal cone dystrophy 3B, 610356 (3), Autosom...   \n",
       "\n",
       "                       How Identified;_x000D_\\nComments Chromosome  \\\n",
       "0     linkage mapping, whole-exome sequencing; a mis...         10   \n",
       "1     deletion mapping, candidate gene; multiple aff...         20   \n",
       "2     linkage mapping, candidate gene; mutations in ...         12   \n",
       "3     homozygosity mapping, candidate gene; symptoms...         20   \n",
       "4     linkage mapping, candidate gene; mutations in ...         12   \n",
       "...                                                 ...        ...   \n",
       "3325  linkage mapping, candidate gene; Criswick-Sche...         11   \n",
       "3326  linkage mapping, candidate gene; may be involv...          1   \n",
       "3327  linkage mapping, candidate gene; Stickler synd...          1   \n",
       "3328  linkage mapping, candidate gene; British famil...          6   \n",
       "3329  homozygosity mapping, candidate gene; mutation...          9   \n",
       "\n",
       "               pLI      exac_pLI   AR   AD AD GOF AD Haploinsuffiency Complex  \n",
       "0     9.145900e-01  5.557300e-01  1.0  1.0    1.0                 0.0     1.0  \n",
       "1     1.000000e+00  1.000000e+00  0.0  1.0    0.0                 1.0     NaN  \n",
       "2     1.000000e+00  1.000000e+00  0.0  1.0    0.0                 1.0     NaN  \n",
       "3     7.403400e-07  1.079200e-02  1.0  0.0    0.0                 0.0     NaN  \n",
       "4     1.000000e+00  1.000000e+00  0.0  1.0    0.0                 1.0     NaN  \n",
       "...            ...           ...  ...  ...    ...                 ...     ...  \n",
       "3325  9.696200e-01  9.207900e-01  0.0  1.0    0.0                 1.0     NaN  \n",
       "3326  5.349900e-48  1.267600e-28  1.0  0.0    0.0                 0.0     NaN  \n",
       "3327  1.000000e+00  9.999800e-01  0.0  1.0    0.0                 1.0     NaN  \n",
       "3328  4.142900e-04  9.918300e-03  0.0  1.0    1.0                 0.0     NaN  \n",
       "3329  1.194400e-26  6.271100e-17  1.0  0.0    0.0                 0.0     NaN  \n",
       "\n",
       "[3330 rows x 154 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Importing RetNet Genes\n",
    "retnet = pd.read_excel(cwd+\"/RetNet/RetNet_AD-AR_FINALIZED.xlsx\", sheet_name=\"gene_info\")\n",
    "retnet.rename(columns={\"gene\": \"Gene.refGene\"}, inplace=True)\n",
    "retnet_genes = pd.Series(retnet[\"Gene.refGene\"])\n",
    "display(retnet.head())\n",
    "\n",
    "# Merging with filtered annotated clinvar data\n",
    "anno_clinvar_filt = pd.merge(anno_clinvar_filt, retnet, on=\"Gene.refGene\", how=\"left\")\n",
    "display(anno_clinvar_filt)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "4843bf89-cfd9-44ce-8df3-5d9af18f72ee",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "BLB    1948\n",
       "PLP    1382\n",
       "Name: PATHOGENICITY, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_clinvar_filt[\"PATHOGENICITY\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "0ade5f2c-afe5-4f17-8d76-01de0a5c011f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Chr', 'Start', 'End', 'Ref', 'Alt', 'QUAL', 'FILTER', 'INFO', 'ID',\n",
       "       'ALLELEID',\n",
       "       ...\n",
       "       'OMIM Phenotypes', 'How Identified;_x000D_\\nComments', 'Chromosome',\n",
       "       'pLI', 'exac_pLI', 'AR', 'AD', 'AD GOF', 'AD Haploinsuffiency',\n",
       "       'Complex'],\n",
       "      dtype='object', length=154)"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "anno_clinvar_filt.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "0d400679-de12-40b3-a420-eaa7e6f54544",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Optional export to csv and excel\n",
    "#anno_clinvar_filt.to_csv(cwd+\"/annovar_files/clinvar_filt_retnet_annotated_PLP-BLB_Nonsynonomous_Variants.csv\", index=False)\n",
    "#anno_clinvar_filt.to_excel(cwd+\"/annovar_files/clinvar_filt_retnet_annotated_PLP-BLB_Nonsynonomous_Variants.xlsx\", index=False) #slow"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "c3bb4984-214a-48a7-9f90-260925793946",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ALL Autosomal Recessive variants = (2392, 154)\n",
      "ONLY Autosomal Recessive variants = (1752, 154)\n",
      "\n",
      "ALL Autosomal Dominant variants = (1524, 154)\n",
      "ONLY Autosomal Dominant variants = (884, 154)\n",
      "\n",
      "ALL Gain-of-Function Autosomal Dominant variants = (761, 154)\n",
      "ONLY Gain-of-Function Autosomal Dominant variants = (156, 154)\n",
      "\n",
      "ALL Haploinsuffient Autosomal Dominant variants = (763, 154)\n",
      "ONLY Haploinsuffient Autosomal Dominant variants = (728, 154)\n"
     ]
    }
   ],
   "source": [
    "# Getting lists of genes with various inheritance patterns\n",
    "ar_all_genes = retnet[retnet[\"AR\"]==1][\"Gene.refGene\"]\n",
    "ar_only_genes = retnet[(retnet[\"AR\"]==1) & (retnet[\"AD\"]==0)][\"Gene.refGene\"]\n",
    "ad_all_genes = retnet[retnet[\"AD\"]==1][\"Gene.refGene\"]\n",
    "ad_only_genes = retnet[(retnet[\"AD\"]==1) & (retnet[\"AR\"]==0)][\"Gene.refGene\"]\n",
    "ad_gof_all_genes = retnet[retnet[\"AD GOF\"]==1][\"Gene.refGene\"]\n",
    "ad_gof_only_genes = retnet[(retnet[\"AD GOF\"]==1) & (retnet[\"AR\"]==0)][\"Gene.refGene\"]\n",
    "ad_haplo_all_genes = retnet[retnet[\"AD Haploinsuffiency\"]==1][\"Gene.refGene\"]\n",
    "ad_haplo_only_genes = retnet[(retnet[\"AD Haploinsuffiency\"]==1) & (retnet[\"AR\"]==0)][\"Gene.refGene\"]\n",
    "\n",
    "# Filting IRD ClinVar variants into different dfs for subsequent ROC + AUC analysis\n",
    "ar_all = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ar_all_genes)]\n",
    "print(\"ALL Autosomal Recessive variants = {}\".format(ar_all.shape))\n",
    "\n",
    "ar_only = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ar_only_genes)]\n",
    "print(\"ONLY Autosomal Recessive variants = {}\".format(ar_only.shape))\n",
    "\n",
    "ad_all = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_all_genes)]\n",
    "print(\"\\nALL Autosomal Dominant variants = {}\".format(ad_all.shape))\n",
    "\n",
    "ad_only = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_only_genes)]\n",
    "print(\"ONLY Autosomal Dominant variants = {}\".format(ad_only.shape))\n",
    "\n",
    "ad_gof_all = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_gof_all_genes)]\n",
    "print(\"\\nALL Gain-of-Function Autosomal Dominant variants = {}\".format(ad_gof_all.shape))\n",
    "\n",
    "ad_gof_only = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_gof_only_genes)]\n",
    "print(\"ONLY Gain-of-Function Autosomal Dominant variants = {}\".format(ad_gof_only.shape))\n",
    "\n",
    "ad_haplo_all = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_haplo_all_genes)]\n",
    "print(\"\\nALL Haploinsuffient Autosomal Dominant variants = {}\".format(ad_haplo_all.shape))\n",
    "\n",
    "ad_haplo_only = anno_clinvar_filt[anno_clinvar_filt[\"Gene.refGene\"].isin(ad_haplo_only_genes)]\n",
    "print(\"ONLY Haploinsuffient Autosomal Dominant variants = {}\".format(ad_haplo_only.shape))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "452e6131-c448-4f5a-9cb5-189fe08875dc",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>index</th>\n",
       "      <th>Gene.refGene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>COL2A1</td>\n",
       "      <td>298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ABCA4</td>\n",
       "      <td>192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>COL11A1</td>\n",
       "      <td>135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ADGRV1</td>\n",
       "      <td>130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NBAS</td>\n",
       "      <td>118</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     index  Gene.refGene\n",
       "0   COL2A1           298\n",
       "1    ABCA4           192\n",
       "2  COL11A1           135\n",
       "3   ADGRV1           130\n",
       "4     NBAS           118"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Number of variants per gene\n",
    "gene_variant_freq = pd.DataFrame(anno_clinvar_filt[\"Gene.refGene\"].value_counts()).reset_index()\n",
    "display(gene_variant_freq.head())\n",
    "#gene_variant_freq.to_excel(cwd+\"/results/gene_variant_frequency.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fb5d4b03-7b8f-494e-a26e-be5852d8d28a",
   "metadata": {},
   "source": [
    "## **Generating ROC and AUC values**"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "82a22d92-c8ac-4c04-ae21-1e0fefbc8143",
   "metadata": {},
   "source": [
    "### ALL Autosomal Recessive IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "fa44ae5b-430e-4310-8ec9-025b1876ab6f",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.96242 while using 96.53% of the values.\n",
      "VEST4_score AUC = 0.93324 while using 96.739% of the values.\n",
      "REVEL_score AUC = 0.95337 while using 96.53% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.89119 while using 93.144% of the values.\n",
      "MutPred_score AUC = 0.89709 while using 63.294% of the values.\n",
      "MVP_score AUC = 0.91972 while using 92.349% of the values.\n",
      "CADD_raw AUC = 0.87433 while using 97.199% of the values.\n",
      "MetaSVM_score AUC = 0.91626 while using 96.53% of the values.\n",
      "MetaRNN_score AUC = 0.98291 while using 97.199% of the values.\n",
      "MetaLR_score AUC = 0.90568 while using 96.53% of the values.\n",
      "M-CAP_score AUC = 0.8994 while using 88.253% of the values.\n",
      "DEOGEN2_score AUC = 0.91722 while using 90.385% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.81636 while using 92.266% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.84152 while using 97.199% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.8931 while using 84.992% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.86219 while using 92.684% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.84522 while using 96.53% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.85405 while using 90.092% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.87595 while using 90.092% of the values.\n",
      "LRT_converted_rankscore AUC = 0.79511 while using 87.918% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.84287 while using 96.656% of the values.\n",
      "MutationAssessor_score AUC = 0.88452 while using 79.933% of the values.\n",
      "MPC_score AUC = 0.82499 while using 85.661% of the values.\n",
      "PrimateAI_score AUC = 0.82148 while using 93.855% of the values.\n",
      "BayesDel_addAF_score AUC = 0.97584 while using 96.823% of the values.\n",
      "BayesDel_noAF_score AUC = 0.94078 while using 96.823% of the values.\n",
      "ClinPred_score AUC = 0.98425 while using 96.614% of the values.\n",
      "LIST-S2_score AUC = 0.84178 while using 94.816% of the values.\n",
      "DANN_score AUC = 0.78727 while using 97.199% of the values.\n",
      "Eigen-raw_coding AUC = 0.91144 while using 85.87% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.89608 while using 85.87% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.74273 while using 97.199% of the values.\n",
      "integrated_fitCons_score AUC = 0.33665 while using 86.413% of the values.\n",
      "GERP++_RS AUC = 0.77006 while using 97.199% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.84771 while using 97.199% of the values.\n",
      "phyloP30way_mammalian AUC = 0.73572 while using 97.199% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.76729 while using 97.199% of the values.\n",
      "phastCons30way_mammalian AUC = 0.6933 while using 97.199% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.7943 while using 97.074% of the values.\n",
      "\n",
      "AUC AR ALL DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.984252</td>\n",
       "      <td>96.613712</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.982913</td>\n",
       "      <td>97.198997</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.975842</td>\n",
       "      <td>96.822742</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.962417</td>\n",
       "      <td>96.530100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.953366</td>\n",
       "      <td>96.530100</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "26        ClinPred_score  0.984252         96.613712\n",
       "8          MetaRNN_score  0.982913         97.198997\n",
       "24  BayesDel_addAF_score  0.975842         96.822742\n",
       "0               MutScore  0.962417         96.530100\n",
       "2            REVEL_score  0.953366         96.530100"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AR ALL DF shape: (21362, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.036417</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.063976</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.094488</td>\n",
       "      <td>0.998</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.122047</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.036417       1.0  MutScore\n",
       "2  0.0  0.063976     0.999  MutScore\n",
       "3  0.0  0.094488     0.998  MutScore\n",
       "4  0.0  0.122047     0.997  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ar_all_data = []\n",
    "roc_ar_all_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ar_all, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ar_all_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ar_all_df = pd.concat([roc_ar_all_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ar_all_df = pd.DataFrame(auc_ar_all_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AR ALL DF shape: {}\".format(auc_ar_all_df.shape))\n",
    "display(auc_ar_all_df.head())\n",
    "print(\"\\nROC AR ALL DF shape: {}\".format(roc_ar_all_df.shape))\n",
    "display(roc_ar_all_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "26a57c58-e076-4397-b887-9a1e20067ccb",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ar_all_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AR_ALL_performance.xlsx\", index=False)\n",
    "#roc_ar_all_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AR_ALL_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e9647d3a-d42f-42eb-a7d7-bb35275cd0ab",
   "metadata": {},
   "source": [
    "### ONLY Autosomal Recessive IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "fb87c345-42ed-4c87-b082-4ad3024210c3",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.95801 while using 95.585% of the values.\n",
      "VEST4_score AUC = 0.93679 while using 95.929% of the values.\n",
      "REVEL_score AUC = 0.95324 while using 95.585% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.88772 while using 91.628% of the values.\n",
      "MutPred_score AUC = 0.8955 while using 58.83% of the values.\n",
      "MVP_score AUC = 0.91828 while using 90.539% of the values.\n",
      "CADD_raw AUC = 0.88489 while using 96.502% of the values.\n",
      "MetaSVM_score AUC = 0.9184 while using 95.585% of the values.\n",
      "MetaRNN_score AUC = 0.98299 while using 96.502% of the values.\n",
      "MetaLR_score AUC = 0.90693 while using 95.585% of the values.\n",
      "M-CAP_score AUC = 0.90041 while using 86.353% of the values.\n",
      "DEOGEN2_score AUC = 0.91381 while using 90.883% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.81743 while using 90.31% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.85267 while using 96.502% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.89918 while using 82.511% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.84969 while using 91.456% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.849 while using 95.7% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.85252 while using 91.342% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.87549 while using 91.342% of the values.\n",
      "LRT_converted_rankscore AUC = 0.79662 while using 89.507% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.83411 while using 96.273% of the values.\n",
      "MutationAssessor_score AUC = 0.87925 while using 85.264% of the values.\n",
      "MPC_score AUC = 0.8136 while using 86.869% of the values.\n",
      "PrimateAI_score AUC = 0.80575 while using 93.635% of the values.\n",
      "BayesDel_addAF_score AUC = 0.97465 while using 96.044% of the values.\n",
      "BayesDel_noAF_score AUC = 0.93974 while using 96.044% of the values.\n",
      "ClinPred_score AUC = 0.98459 while using 95.757% of the values.\n",
      "LIST-S2_score AUC = 0.82265 while using 96.101% of the values.\n",
      "DANN_score AUC = 0.79105 while using 96.502% of the values.\n",
      "Eigen-raw_coding AUC = 0.91692 while using 82.913% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.90517 while using 82.913% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.72497 while using 96.502% of the values.\n",
      "integrated_fitCons_score AUC = 0.30202 while using 83.658% of the values.\n",
      "GERP++_RS AUC = 0.77619 while using 96.502% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.85656 while using 96.502% of the values.\n",
      "phyloP30way_mammalian AUC = 0.74686 while using 96.502% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.76069 while using 96.502% of the values.\n",
      "phastCons30way_mammalian AUC = 0.6881 while using 96.502% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.80222 while using 96.33% of the values.\n",
      "\n",
      "AUC AR ONLY DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.984590</td>\n",
       "      <td>95.756881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.982994</td>\n",
       "      <td>96.502294</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.974650</td>\n",
       "      <td>96.043578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.958007</td>\n",
       "      <td>95.584862</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.953237</td>\n",
       "      <td>95.584862</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "26        ClinPred_score  0.984590         95.756881\n",
       "8          MetaRNN_score  0.982994         96.502294\n",
       "24  BayesDel_addAF_score  0.974650         96.043578\n",
       "0               MutScore  0.958007         95.584862\n",
       "2            REVEL_score  0.953237         95.584862"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AR ONLY DF shape: (17046, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.034175</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.057949</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.10847</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.132244</td>\n",
       "      <td>0.996</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.034175       1.0  MutScore\n",
       "2  0.0  0.057949     0.999  MutScore\n",
       "3  0.0   0.10847     0.997  MutScore\n",
       "4  0.0  0.132244     0.996  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ar_only_data = []\n",
    "roc_ar_only_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ar_only, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ar_only_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ar_only_df = pd.concat([roc_ar_only_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ar_only_df = pd.DataFrame(auc_ar_only_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AR ONLY DF shape: {}\".format(auc_ar_only_df.shape))\n",
    "display(auc_ar_only_df.head())\n",
    "print(\"\\nROC AR ONLY DF shape: {}\".format(roc_ar_only_df.shape))\n",
    "display(roc_ar_only_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "826657b2-39e9-4a8c-88cf-2bc92f8963df",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ar_only_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AR_ONLY_performance.xlsx\", index=False)\n",
    "#roc_ar_only_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AR_ONLY_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3e0fa969-d25a-4e42-80fd-8f428be5aae6",
   "metadata": {},
   "source": [
    "### ALL Autosomal Dominant (GOF + Haploinsuffiency + AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "a360f320-33bd-426d-a04d-7dcb226ce0b7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.96883 while using 99.147% of the values.\n",
      "VEST4_score AUC = 0.92845 while using 99.081% of the values.\n",
      "REVEL_score AUC = 0.94261 while using 99.147% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.88802 while using 98.36% of the values.\n",
      "MutPred_score AUC = 0.92564 while using 75.787% of the values.\n",
      "MVP_score AUC = 0.93086 while using 97.835% of the values.\n",
      "CADD_raw AUC = 0.84611 while using 99.147% of the values.\n",
      "MetaSVM_score AUC = 0.89983 while using 99.147% of the values.\n",
      "MetaRNN_score AUC = 0.96832 while using 99.147% of the values.\n",
      "MetaLR_score AUC = 0.89867 while using 99.147% of the values.\n",
      "M-CAP_score AUC = 0.92198 while using 95.079% of the values.\n",
      "DEOGEN2_score AUC = 0.9074 while using 92.126% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.82305 while using 98.491% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.77145 while using 99.147% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.84277 while using 95.997% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.89218 while using 97.835% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.87077 while using 98.819% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.84682 while using 93.438% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.87894 while using 93.438% of the values.\n",
      "LRT_converted_rankscore AUC = 0.76694 while using 91.864% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.78799 while using 98.556% of the values.\n",
      "MutationAssessor_score AUC = 0.91139 while using 83.005% of the values.\n",
      "MPC_score AUC = 0.81138 while using 87.008% of the values.\n",
      "PrimateAI_score AUC = 0.83819 while using 96.785% of the values.\n",
      "BayesDel_addAF_score AUC = 0.96211 while using 99.081% of the values.\n",
      "BayesDel_noAF_score AUC = 0.93544 while using 99.081% of the values.\n",
      "ClinPred_score AUC = 0.96773 while using 99.081% of the values.\n",
      "LIST-S2_score AUC = 0.85746 while using 95.866% of the values.\n",
      "DANN_score AUC = 0.75435 while using 99.147% of the values.\n",
      "Eigen-raw_coding AUC = 0.89258 while using 96.916% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.8648 while using 96.916% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.7355 while using 99.147% of the values.\n",
      "integrated_fitCons_score AUC = 0.4202 while using 96.916% of the values.\n",
      "GERP++_RS AUC = 0.64441 while using 99.147% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.77466 while using 99.147% of the values.\n",
      "phyloP30way_mammalian AUC = 0.58465 while using 99.147% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.69988 while using 99.147% of the values.\n",
      "phastCons30way_mammalian AUC = 0.60884 while using 99.147% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.71914 while using 99.147% of the values.\n",
      "\n",
      "AUC AD ALL DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.968828</td>\n",
       "      <td>99.146982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.968321</td>\n",
       "      <td>99.146982</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.967731</td>\n",
       "      <td>99.081365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.962108</td>\n",
       "      <td>99.081365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.942611</td>\n",
       "      <td>99.146982</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "0               MutScore  0.968828         99.146982\n",
       "8          MetaRNN_score  0.968321         99.146982\n",
       "26        ClinPred_score  0.967731         99.081365\n",
       "24  BayesDel_addAF_score  0.962108         99.081365\n",
       "2            REVEL_score  0.942611         99.146982"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD ALL DF shape: (15235, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.040172</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.077475</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.13056</td>\n",
       "      <td>0.998</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.177905</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.040172       1.0  MutScore\n",
       "2  0.0  0.077475     0.999  MutScore\n",
       "3  0.0   0.13056     0.998  MutScore\n",
       "4  0.0  0.177905     0.997  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_all_data = []\n",
    "roc_ad_all_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_all, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_all_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_all_df = pd.concat([roc_ad_all_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_all_df = pd.DataFrame(auc_ad_all_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD ALL DF shape: {}\".format(auc_ad_all_df.shape))\n",
    "display(auc_ad_all_df.head())\n",
    "print(\"\\nROC AD ALL DF shape: {}\".format(roc_ad_all_df.shape))\n",
    "display(roc_ad_all_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "c77eb929-892e-4d8e-9969-ea11e79113f4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_all_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_ALL_performance.xlsx\", index=False)\n",
    "#roc_ad_all_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_ALL_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "35ecf7f9-3479-4825-a484-ac88a44085ce",
   "metadata": {},
   "source": [
    "### ONLY Autosomal Dominant (GOF + Haploinsuffiency with NO AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "72086665-3434-4aef-b779-83c95fc92d52",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.9713 while using 99.208% of the values.\n",
      "VEST4_score AUC = 0.94309 while using 99.208% of the values.\n",
      "REVEL_score AUC = 0.9618 while using 99.208% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.88406 while using 99.208% of the values.\n",
      "MutPred_score AUC = 0.94521 while using 76.244% of the values.\n",
      "MVP_score AUC = 0.94857 while using 98.303% of the values.\n",
      "CADD_raw AUC = 0.87664 while using 99.208% of the values.\n",
      "MetaSVM_score AUC = 0.91595 while using 99.208% of the values.\n",
      "MetaRNN_score AUC = 0.96994 while using 99.208% of the values.\n",
      "MetaLR_score AUC = 0.93082 while using 99.208% of the values.\n",
      "M-CAP_score AUC = 0.9473 while using 96.38% of the values.\n",
      "DEOGEN2_score AUC = 0.92846 while using 94.457% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.88163 while using 99.208% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.78146 while using 99.208% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.84714 while using 99.208% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.90173 while using 99.208% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.89961 while using 98.869% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.84551 while using 98.416% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.88996 while using 98.416% of the values.\n",
      "LRT_converted_rankscore AUC = 0.78893 while using 97.964% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.7655 while using 99.208% of the values.\n",
      "MutationAssessor_score AUC = 0.93155 while using 95.928% of the values.\n",
      "MPC_score AUC = 0.81812 while using 90.498% of the values.\n",
      "PrimateAI_score AUC = 0.93311 while using 98.529% of the values.\n",
      "BayesDel_addAF_score AUC = 0.9744 while using 99.208% of the values.\n",
      "BayesDel_noAF_score AUC = 0.95578 while using 99.208% of the values.\n",
      "ClinPred_score AUC = 0.97215 while using 99.208% of the values.\n",
      "LIST-S2_score AUC = 0.86289 while using 99.208% of the values.\n",
      "DANN_score AUC = 0.73836 while using 99.208% of the values.\n",
      "Eigen-raw_coding AUC = 0.9137 while using 99.208% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.89054 while using 99.208% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.71958 while using 99.208% of the values.\n",
      "integrated_fitCons_score AUC = 0.43643 while using 99.208% of the values.\n",
      "GERP++_RS AUC = 0.57961 while using 99.208% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.7564 while using 99.208% of the values.\n",
      "phyloP30way_mammalian AUC = 0.48276 while using 99.208% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.65634 while using 99.208% of the values.\n",
      "phastCons30way_mammalian AUC = 0.5816 while using 99.208% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.70439 while using 99.208% of the values.\n",
      "\n",
      "AUC AD ONLY DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.974398</td>\n",
       "      <td>99.208145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.972150</td>\n",
       "      <td>99.208145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.971298</td>\n",
       "      <td>99.208145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.969937</td>\n",
       "      <td>99.208145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.961801</td>\n",
       "      <td>99.208145</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "24  BayesDel_addAF_score  0.974398         99.208145\n",
       "26        ClinPred_score  0.972150         99.208145\n",
       "0               MutScore  0.971298         99.208145\n",
       "8          MetaRNN_score  0.969937         99.208145\n",
       "2            REVEL_score  0.961801         99.208145"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD ONLY DF shape: (8840, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.039326</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.078652</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.143258</td>\n",
       "      <td>0.998</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.205056</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.039326       1.0  MutScore\n",
       "2  0.0  0.078652     0.999  MutScore\n",
       "3  0.0  0.143258     0.998  MutScore\n",
       "4  0.0  0.205056     0.997  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_only_data = []\n",
    "roc_ad_only_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_only, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_only_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_only_df = pd.concat([roc_ad_only_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_only_df = pd.DataFrame(auc_ad_only_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD ONLY DF shape: {}\".format(auc_ad_only_df.shape))\n",
    "display(auc_ad_only_df.head())\n",
    "print(\"\\nROC AD ONLY DF shape: {}\".format(roc_ad_only_df.shape))\n",
    "display(roc_ad_only_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "1e269585-4d67-48f8-b308-632eadfc9b62",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_only_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_ONLY_performance.xlsx\", index=False)\n",
    "#roc_ad_only_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_ONLY_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3df55672-822a-403d-ae9a-c7bdac9e162c",
   "metadata": {},
   "source": [
    "### ALL GOF Autosomal Dominant (with AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "77252119-a20a-406a-b02a-15c915d2d33e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.97222 while using 99.08% of the values.\n",
      "VEST4_score AUC = 0.92351 while using 98.949% of the values.\n",
      "REVEL_score AUC = 0.94895 while using 99.08% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.89337 while using 97.503% of the values.\n",
      "MutPred_score AUC = 0.88684 while using 68.2% of the values.\n",
      "MVP_score AUC = 0.91504 while using 96.583% of the values.\n",
      "CADD_raw AUC = 0.83245 while using 99.08% of the values.\n",
      "MetaSVM_score AUC = 0.91251 while using 99.08% of the values.\n",
      "MetaRNN_score AUC = 0.9854 while using 99.08% of the values.\n",
      "MetaLR_score AUC = 0.90866 while using 99.08% of the values.\n",
      "M-CAP_score AUC = 0.90462 while using 91.459% of the values.\n",
      "DEOGEN2_score AUC = 0.9278 while using 86.202% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.82988 while using 97.766% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.79884 while using 99.08% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.86836 while using 96.189% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.87888 while using 97.503% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.82352 while using 98.817% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.83173 while using 88.83% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.84785 while using 88.83% of the values.\n",
      "LRT_converted_rankscore AUC = 0.76803 while using 86.071% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.83977 while using 97.898% of the values.\n",
      "MutationAssessor_score AUC = 0.88411 while using 70.565% of the values.\n",
      "MPC_score AUC = 0.85435 while using 80.289% of the values.\n",
      "PrimateAI_score AUC = 0.85943 while using 95.138% of the values.\n",
      "BayesDel_addAF_score AUC = 0.98395 while using 98.949% of the values.\n",
      "BayesDel_noAF_score AUC = 0.94514 while using 98.949% of the values.\n",
      "ClinPred_score AUC = 0.9827 while using 98.949% of the values.\n",
      "LIST-S2_score AUC = 0.88753 while using 92.51% of the values.\n",
      "DANN_score AUC = 0.75285 while using 99.08% of the values.\n",
      "Eigen-raw_coding AUC = 0.88749 while using 98.029% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.86543 while using 98.029% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.76328 while using 99.08% of the values.\n",
      "integrated_fitCons_score AUC = 0.44273 while using 98.029% of the values.\n",
      "GERP++_RS AUC = 0.73216 while using 99.08% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.81257 while using 99.08% of the values.\n",
      "phyloP30way_mammalian AUC = 0.7063 while using 99.08% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.76109 while using 99.08% of the values.\n",
      "phastCons30way_mammalian AUC = 0.68936 while using 99.08% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.76034 while using 99.08% of the values.\n",
      "\n",
      "AUC AD GOF ALL DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.985403</td>\n",
       "      <td>99.080158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.983950</td>\n",
       "      <td>98.948752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.982704</td>\n",
       "      <td>98.948752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.972220</td>\n",
       "      <td>99.080158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.948948</td>\n",
       "      <td>99.080158</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "8          MetaRNN_score  0.985403         99.080158\n",
       "24  BayesDel_addAF_score  0.983950         98.948752\n",
       "26        ClinPred_score  0.982704         98.948752\n",
       "0               MutScore  0.972220         99.080158\n",
       "2            REVEL_score  0.948948         99.080158"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD GOF ALL DF shape: (8672, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.040541</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.072973</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.113514</td>\n",
       "      <td>0.998</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.143243</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.040541       1.0  MutScore\n",
       "2  0.0  0.072973     0.999  MutScore\n",
       "3  0.0  0.113514     0.998  MutScore\n",
       "4  0.0  0.143243     0.997  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_gof_all_data = []\n",
    "roc_ad_gof_all_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_gof_all, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_gof_all_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_gof_all_df = pd.concat([roc_ad_gof_all_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_gof_all_df = pd.DataFrame(auc_ad_gof_all_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD GOF ALL DF shape: {}\".format(auc_ad_gof_all_df.shape))\n",
    "display(auc_ad_gof_all_df.head())\n",
    "print(\"\\nROC AD GOF ALL DF shape: {}\".format(roc_ad_gof_all_df.shape))\n",
    "display(roc_ad_gof_all_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "a9ccb3d4-640a-4ede-8842-654789c11192",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_gof_all_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_GOF_ALL_performance.xlsx\", index=False)\n",
    "#roc_ad_gof_all_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_GOF_ALL_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "41bb7e74-bfb6-4ce7-af4e-c4233b101b3a",
   "metadata": {},
   "source": [
    "### ONLY GOF Autosomal Dominant (with NO AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "cfad90b0-aa01-4909-a38f-01269d1aecad",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.97234 while using 99.359% of the values.\n",
      "VEST4_score AUC = 0.94293 while using 99.359% of the values.\n",
      "REVEL_score AUC = 0.96394 while using 99.359% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.87255 while using 99.359% of the values.\n",
      "MutPred_score AUC = 0.88312 while using 43.59% of the values.\n",
      "MVP_score AUC = 0.92005 while using 94.231% of the values.\n",
      "CADD_raw AUC = 0.8436 while using 99.359% of the values.\n",
      "MetaSVM_score AUC = 0.93184 while using 99.359% of the values.\n",
      "MetaRNN_score AUC = 0.99323 while using 99.359% of the values.\n",
      "MetaLR_score AUC = 0.93651 while using 99.359% of the values.\n",
      "M-CAP_score AUC = 0.91199 while using 83.974% of the values.\n",
      "DEOGEN2_score AUC = 0.93465 while using 73.077% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.90966 while using 99.359% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.82026 while using 99.359% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.89379 while using 99.359% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.875 while using 99.359% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.85387 while using 99.359% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.79412 while using 95.513% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.79629 while using 95.513% of the values.\n",
      "LRT_converted_rankscore AUC = 0.83932 while using 97.436% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.81723 while using 99.359% of the values.\n",
      "MutationAssessor_score AUC = 0.86882 while using 94.231% of the values.\n",
      "MPC_score AUC = 0.92875 while using 75.641% of the values.\n",
      "PrimateAI_score AUC = 0.93519 while using 98.077% of the values.\n",
      "BayesDel_addAF_score AUC = 0.99732 while using 99.359% of the values.\n",
      "BayesDel_noAF_score AUC = 0.96429 while using 99.359% of the values.\n",
      "ClinPred_score AUC = 0.99125 while using 99.359% of the values.\n",
      "LIST-S2_score AUC = 0.88644 while using 99.359% of the values.\n",
      "DANN_score AUC = 0.76564 while using 99.359% of the values.\n",
      "Eigen-raw_coding AUC = 0.88994 while using 99.359% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.88329 while using 99.359% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.72607 while using 99.359% of the values.\n",
      "integrated_fitCons_score AUC = 0.64064 while using 99.359% of the values.\n",
      "GERP++_RS AUC = 0.7423 while using 99.359% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.83263 while using 99.359% of the values.\n",
      "phyloP30way_mammalian AUC = 0.76926 while using 99.359% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.74253 while using 99.359% of the values.\n",
      "phastCons30way_mammalian AUC = 0.81769 while using 99.359% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.78198 while using 99.359% of the values.\n",
      "\n",
      "AUC AD GOF ONLY DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.997316</td>\n",
       "      <td>99.358974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.993231</td>\n",
       "      <td>99.358974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.991246</td>\n",
       "      <td>99.358974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.972339</td>\n",
       "      <td>99.358974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>BayesDel_noAF_score</td>\n",
       "      <td>0.964286</td>\n",
       "      <td>99.358974</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "24  BayesDel_addAF_score  0.997316         99.358974\n",
       "8          MetaRNN_score  0.993231         99.358974\n",
       "26        ClinPred_score  0.991246         99.358974\n",
       "0               MutScore  0.972339         99.358974\n",
       "25   BayesDel_noAF_score  0.964286         99.358974"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD GOF ONLY DF shape: (1564, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.027778</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.194444</td>\n",
       "      <td>0.993</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.25</td>\n",
       "      <td>0.991</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>0.973</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.027778       1.0  MutScore\n",
       "2  0.0  0.194444     0.993  MutScore\n",
       "3  0.0      0.25     0.991  MutScore\n",
       "4  0.0  0.333333     0.973  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_gof_only_data = []\n",
    "roc_ad_gof_only_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_gof_only, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_gof_only_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_gof_only_df = pd.concat([roc_ad_gof_only_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_gof_only_df = pd.DataFrame(auc_ad_gof_only_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD GOF ONLY DF shape: {}\".format(auc_ad_gof_only_df.shape))\n",
    "display(auc_ad_gof_only_df.head())\n",
    "print(\"\\nROC AD GOF ONLY DF shape: {}\".format(roc_ad_gof_only_df.shape))\n",
    "display(roc_ad_gof_only_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "062f7519-16df-4784-9828-b45f9121d515",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_gof_only_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_GOF_ONLY_performance.xlsx\", index=False)\n",
    "#roc_ad_gof_only_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_GOF_ONLY_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2832adb2-44e4-44cd-882c-c5d280e375b4",
   "metadata": {},
   "source": [
    "### ALL Haploinsuffiency Autosomal Dominant (with AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "652c0103-e2e0-4df1-854a-d950a43a3721",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.97252 while using 99.214% of the values.\n",
      "VEST4_score AUC = 0.94246 while using 99.214% of the values.\n",
      "REVEL_score AUC = 0.96242 while using 99.214% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.88922 while using 99.214% of the values.\n",
      "MutPred_score AUC = 0.95069 while using 83.355% of the values.\n",
      "MVP_score AUC = 0.95611 while using 99.083% of the values.\n",
      "CADD_raw AUC = 0.88468 while using 99.214% of the values.\n",
      "MetaSVM_score AUC = 0.91798 while using 99.214% of the values.\n",
      "MetaRNN_score AUC = 0.9653 while using 99.214% of the values.\n",
      "MetaLR_score AUC = 0.9354 while using 99.214% of the values.\n",
      "M-CAP_score AUC = 0.9546 while using 98.689% of the values.\n",
      "DEOGEN2_score AUC = 0.92934 while using 98.034% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.88797 while using 99.214% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.77838 while using 99.214% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.83578 while using 95.806% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.90912 while using 98.165% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.91993 while using 98.82% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.863 while using 98.034% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.91194 while using 98.034% of the values.\n",
      "LRT_converted_rankscore AUC = 0.78131 while using 97.641% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.75753 while using 99.214% of the values.\n",
      "MutationAssessor_score AUC = 0.93944 while using 95.413% of the values.\n",
      "MPC_score AUC = 0.79612 while using 93.709% of the values.\n",
      "PrimateAI_score AUC = 0.91843 while using 98.427% of the values.\n",
      "BayesDel_addAF_score AUC = 0.97171 while using 99.214% of the values.\n",
      "BayesDel_noAF_score AUC = 0.95532 while using 99.214% of the values.\n",
      "ClinPred_score AUC = 0.97017 while using 99.214% of the values.\n",
      "LIST-S2_score AUC = 0.8594 while using 99.214% of the values.\n",
      "DANN_score AUC = 0.75756 while using 99.214% of the values.\n",
      "Eigen-raw_coding AUC = 0.91484 while using 95.806% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.88804 while using 95.806% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.71837 while using 99.214% of the values.\n",
      "integrated_fitCons_score AUC = 0.37317 while using 95.806% of the values.\n",
      "GERP++_RS AUC = 0.56126 while using 99.214% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.74411 while using 99.214% of the values.\n",
      "phyloP30way_mammalian AUC = 0.43845 while using 99.214% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.64994 while using 99.214% of the values.\n",
      "phastCons30way_mammalian AUC = 0.55278 while using 99.214% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.68871 while using 99.214% of the values.\n",
      "\n",
      "AUC AD HAPLO ALL DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.972520</td>\n",
       "      <td>99.21363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.971709</td>\n",
       "      <td>99.21363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.970173</td>\n",
       "      <td>99.21363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.965305</td>\n",
       "      <td>99.21363</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.962424</td>\n",
       "      <td>99.21363</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "0               MutScore  0.972520          99.21363\n",
       "24  BayesDel_addAF_score  0.971709          99.21363\n",
       "26        ClinPred_score  0.970173          99.21363\n",
       "8          MetaRNN_score  0.965305          99.21363\n",
       "2            REVEL_score  0.962424          99.21363"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD HAPLO ALL DF shape: (7476, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.039755</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.082569</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.217125</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.002326</td>\n",
       "      <td>0.269113</td>\n",
       "      <td>0.996</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        fpr       tpr threshold     model\n",
       "0       0.0       0.0       2.0  MutScore\n",
       "1       0.0  0.039755       1.0  MutScore\n",
       "2       0.0  0.082569     0.999  MutScore\n",
       "3       0.0  0.217125     0.997  MutScore\n",
       "4  0.002326  0.269113     0.996  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_haplo_all_data = []\n",
    "roc_ad_haplo_all_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_haplo_all, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_haplo_all_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_haplo_all_df = pd.concat([roc_ad_haplo_all_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_haplo_all_df = pd.DataFrame(auc_ad_haplo_all_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD HAPLO ALL DF shape: {}\".format(auc_ad_haplo_all_df.shape))\n",
    "display(auc_ad_haplo_all_df.head())\n",
    "print(\"\\nROC AD HAPLO ALL DF shape: {}\".format(roc_ad_haplo_all_df.shape))\n",
    "display(roc_ad_haplo_all_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "93848153-62e7-42c8-8c48-8a054a331215",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_haplo_all_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_HAPLO_ALL_performance.xlsx\", index=False)\n",
    "#roc_ad_haplo_all_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_HAPLO_ALL_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "697aadb2-3cc7-4bd6-9e9a-e0df34f01965",
   "metadata": {},
   "source": [
    "### ONLY Haploinsuffiency Autosomal Dominant (with NO AR Overlap) IRD SNVs Performance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "db883d58-d1dc-4528-b3c3-3db946f86ecb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MutScore AUC = 0.9707 while using 99.176% of the values.\n",
      "VEST4_score AUC = 0.94096 while using 99.176% of the values.\n",
      "REVEL_score AUC = 0.96054 while using 99.176% of the values.\n",
      "PROVEAN_converted_rankscore AUC = 0.88238 while using 99.176% of the values.\n",
      "MutPred_score AUC = 0.94899 while using 83.242% of the values.\n",
      "MVP_score AUC = 0.95349 while using 99.176% of the values.\n",
      "CADD_raw AUC = 0.87839 while using 99.176% of the values.\n",
      "MetaSVM_score AUC = 0.91298 while using 99.176% of the values.\n",
      "MetaRNN_score AUC = 0.96393 while using 99.176% of the values.\n",
      "MetaLR_score AUC = 0.93169 while using 99.176% of the values.\n",
      "M-CAP_score AUC = 0.95481 while using 99.038% of the values.\n",
      "DEOGEN2_score AUC = 0.92972 while using 99.038% of the values.\n",
      "FATHMM_converted_rankscore AUC = 0.88308 while using 99.176% of the values.\n",
      "fathmm-MKL_coding_score AUC = 0.76377 while using 99.176% of the values.\n",
      "fathmm-XF_coding_score AUC = 0.83276 while using 99.176% of the values.\n",
      "SIFT_converted_rankscore AUC = 0.90548 while using 99.176% of the values.\n",
      "SIFT4G_converted_rankscore AUC = 0.9155 while using 98.764% of the values.\n",
      "Polyphen2_HDIV_score AUC = 0.85877 while using 99.038% of the values.\n",
      "Polyphen2_HVAR_score AUC = 0.90937 while using 99.038% of the values.\n",
      "LRT_converted_rankscore AUC = 0.77123 while using 98.077% of the values.\n",
      "MutationTaster_converted_rankscore AUC = 0.74555 while using 99.176% of the values.\n",
      "MutationAssessor_score AUC = 0.93821 while using 96.291% of the values.\n",
      "MPC_score AUC = 0.80304 while using 93.681% of the values.\n",
      "PrimateAI_score AUC = 0.92893 while using 98.626% of the values.\n",
      "BayesDel_addAF_score AUC = 0.97177 while using 99.176% of the values.\n",
      "BayesDel_noAF_score AUC = 0.95431 while using 99.176% of the values.\n",
      "ClinPred_score AUC = 0.96795 while using 99.176% of the values.\n",
      "LIST-S2_score AUC = 0.85191 while using 99.176% of the values.\n",
      "DANN_score AUC = 0.7429 while using 99.176% of the values.\n",
      "Eigen-raw_coding AUC = 0.91367 while using 99.176% of the values.\n",
      "Eigen-PC-raw_coding AUC = 0.88623 while using 99.176% of the values.\n",
      "GenoCanyon_rankscore AUC = 0.70948 while using 99.176% of the values.\n",
      "integrated_fitCons_score AUC = 0.37459 while using 99.176% of the values.\n",
      "GERP++_RS AUC = 0.53076 while using 99.176% of the values.\n",
      "phyloP100way_vertebrate AUC = 0.72835 while using 99.176% of the values.\n",
      "phyloP30way_mammalian AUC = 0.4125 while using 99.176% of the values.\n",
      "phastCons100way_vertebrate AUC = 0.62933 while using 99.176% of the values.\n",
      "phastCons30way_mammalian AUC = 0.52896 while using 99.176% of the values.\n",
      "SiPhy_29way_logOdds AUC = 0.66675 while using 99.176% of the values.\n",
      "\n",
      "AUC AD HAPLO ONLY DF shape: (39, 3)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Model</th>\n",
       "      <th>AUC</th>\n",
       "      <th>Percent_Retained</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>BayesDel_addAF_score</td>\n",
       "      <td>0.971774</td>\n",
       "      <td>99.175824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MutScore</td>\n",
       "      <td>0.970697</td>\n",
       "      <td>99.175824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ClinPred_score</td>\n",
       "      <td>0.967949</td>\n",
       "      <td>99.175824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>MetaRNN_score</td>\n",
       "      <td>0.963926</td>\n",
       "      <td>99.175824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>REVEL_score</td>\n",
       "      <td>0.960537</td>\n",
       "      <td>99.175824</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   Model       AUC  Percent_Retained\n",
       "24  BayesDel_addAF_score  0.971774         99.175824\n",
       "0               MutScore  0.970697         99.175824\n",
       "26        ClinPred_score  0.967949         99.175824\n",
       "8          MetaRNN_score  0.963926         99.175824\n",
       "2            REVEL_score  0.960537         99.175824"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "ROC AD HAPLO ONLY DF shape: (7187, 4)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fpr</th>\n",
       "      <th>tpr</th>\n",
       "      <th>threshold</th>\n",
       "      <th>model</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.040625</td>\n",
       "      <td>1.0</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.084375</td>\n",
       "      <td>0.999</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.153125</td>\n",
       "      <td>0.998</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.21875</td>\n",
       "      <td>0.997</td>\n",
       "      <td>MutScore</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   fpr       tpr threshold     model\n",
       "0  0.0       0.0       2.0  MutScore\n",
       "1  0.0  0.040625       1.0  MutScore\n",
       "2  0.0  0.084375     0.999  MutScore\n",
       "3  0.0  0.153125     0.998  MutScore\n",
       "4  0.0   0.21875     0.997  MutScore"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of models to calculate ROC & AUC values\n",
    "models = [\"MutScore\", \"VEST4_score\", \"REVEL_score\", \"PROVEAN_converted_rankscore\", \"MutPred_score\", \"MVP_score\", \"CADD_raw\", \n",
    "          \"MetaSVM_score\", \"MetaRNN_score\", \"MetaLR_score\", \"M-CAP_score\", \"DEOGEN2_score\", \"FATHMM_converted_rankscore\", \n",
    "          \"fathmm-MKL_coding_score\", \"fathmm-XF_coding_score\", \"SIFT_converted_rankscore\", \"SIFT4G_converted_rankscore\", \n",
    "          \"Polyphen2_HDIV_score\", \"Polyphen2_HVAR_score\", \"LRT_converted_rankscore\", \"MutationTaster_converted_rankscore\", \n",
    "          \"MutationAssessor_score\", \"MPC_score\", \"PrimateAI_score\", \"BayesDel_addAF_score\", \"BayesDel_noAF_score\", \n",
    "          \"ClinPred_score\", \"LIST-S2_score\", \"DANN_score\", \"Eigen-raw_coding\", \"Eigen-PC-raw_coding\", \"GenoCanyon_rankscore\", \n",
    "          \"integrated_fitCons_score\", \"GERP++_RS\", \"phyloP100way_vertebrate\", \"phyloP30way_mammalian\", \"phastCons100way_vertebrate\", \n",
    "          \"phastCons30way_mammalian\", \"SiPhy_29way_logOdds\"]\n",
    "\n",
    "# Looping and creating lists/dfs to append results to\n",
    "auc_ad_haplo_only_data = []\n",
    "roc_ad_haplo_only_df = pd.DataFrame(columns=[\"fpr\", \"tpr\", \"threshold\", \"model\"])\n",
    "for m in models:\n",
    "    # Calulating parametrics for each prediction model\n",
    "    fpr, tpr, threshold, auc_score, percent_retained = roc_auc(df=ad_haplo_only, model=m, truth=\"y_test\", disp_results=True)\n",
    "    \n",
    "    # Appending a list with a tuple containing model name, its AUC score, and its percent SNV retained\n",
    "    auc_ad_haplo_only_data.append((m, auc_score, percent_retained))\n",
    "    \n",
    "    # Making a dataframe with fpr, tpr, thresholds, and model ID for drawing ROC curves\n",
    "    model_df = pd.DataFrame({\"fpr\": fpr, \"tpr\": tpr, \"threshold\": threshold, \"model\": np.full(len(fpr), m)})\n",
    "    roc_ad_haplo_only_df = pd.concat([roc_ad_haplo_only_df, model_df], axis=0)\n",
    "\n",
    "# Making the AUC dataframe from list of tuples\n",
    "auc_ad_haplo_only_df = pd.DataFrame(auc_ad_haplo_only_data, columns=[\"Model\", \"AUC\", \"Percent_Retained\"]).sort_values(\"AUC\", ascending=False)\n",
    "\n",
    "# Displaying results\n",
    "print(\"\\nAUC AD HAPLO ONLY DF shape: {}\".format(auc_ad_haplo_only_df.shape))\n",
    "display(auc_ad_haplo_only_df.head())\n",
    "print(\"\\nROC AD HAPLO ONLY DF shape: {}\".format(roc_ad_haplo_only_df.shape))\n",
    "display(roc_ad_haplo_only_df.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "2ed4df0e-c9e4-4c9f-98a9-d4c83a1748ad",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Exporting to graph in R\n",
    "#auc_ad_haplo_only_df.to_excel(cwd+\"/AUC_ROC_results/auc/AUC_AD_HAPLO_ONLY_performance.xlsx\", index=False)\n",
    "#roc_ad_haplo_only_df.to_excel(cwd+\"/AUC_ROC_results/roc/ROC_AD_HAPLO_ONLY_performance.xlsx\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "69aef8f2-e14a-439f-9b2c-40c6e955c99d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fdd80cfc-8cc5-41c3-9ebe-698191c609cc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "81f94932-a998-46c7-b333-57447074b6e2",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cd96a8c5-5273-4d2f-9bec-e5992acf2725",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ab08c9f-d115-49e4-9a2d-f5593390b1e9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
